// agency-frontend/src/pages/agency/AgencyOffers.tsx
import { useState, useEffect } from "react";
import { useParams } from "react-router-dom"; 
import { AgencyLayout } from "@/components/agency/AgencyLayout";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import { Textarea } from "@/components/ui/textarea";
import { Dialog, DialogContent, DialogDescription, DialogFooter, DialogHeader, DialogTitle, DialogTrigger } from "@/components/ui/dialog";
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
import { ImageUpload } from "@/components/ui/image-upload";
import { MultipleImageUpload } from "@/components/ui/multiple-image-upload";
import { Search, Plus, Filter, ShoppingCart, Clock, CheckCircle, Edit } from "lucide-react";
import { useToast } from "@/hooks/use-toast";
// Ligne 15 (après les imports existants)
import { useTranslation } from "react-i18next";
import { useAuth } from "@/context/AuthContext";
import { Link, useNavigate } from "react-router-dom";

// ✅ Type pour une offre
type Offer = {
  id: number;
  title: string;
  description: string;
  price: string;
  currency?: string;
  category: string;
  validUntil?: string;
  delivery_date?: string;
  avatar?: string;
  images?: string[]; // Nouvelles images multiples
  seller_profile?: string;
  condition_state?: "new" | "like_new" | "good" | "fair" | "poor" | "to_repair";
  tags?: string[];
  reviews?: Array<{
    rating: number;
    comment: string;
    user_name?: string;
  }>;
  status: "active" | "pending" | "unknown";
  clientsCount: number;
};

// ✅ Fonction pour obtenir le badge de statut
const getStatusBadge = (status: Offer["status"], t: any) => {
  switch (status) {
    case "active":
      return (
        <Badge className="bg-success/10 text-success border-success/20">
          <CheckCircle className="w-3 h-3 mr-1" />{t("active")}
        </Badge>
      );
    case "pending":
      return (
        <Badge variant="secondary">
          <Clock className="w-3 h-3 mr-1" />{t("common.pending")}
        </Badge>
      );
    default:
      return <Badge variant="outline">{t("common.unknown")}</Badge>;
  }
};

//composant!!!
export default function AgencyOffers() {
  //ajout pour la traduction
  const { t } = useTranslation(["offers", "common"]);
  const { toast } = useToast();
  const { id } = useParams();
  const { hasPermission, isAdmin, getIsLoading } = useAuth();
  const navigate = useNavigate();

  // ✅ State typés
  const [offers, setOffers] = useState<Offer[]>([]);
  const [isDialogOpen, setIsDialogOpen] = useState(false);
  const [newOffer, setNewOffer] = useState<Omit<Offer, "id" | "status" | "clientsCount">>({
    title: "",
    description: "",
    price: "",
    currency: "EUR",
    category: "",
    validUntil: "",
    delivery_date: "",
    avatar: "",
    images: [], // Nouveau champ pour images multiples
    condition_state: "new",
    tags: [],
    reviews: [],
  });

  const [offerToDelete, setOfferToDelete] = useState<Offer | null>(null);
  const [isCustomCategory, setIsCustomCategory] = useState(false);
  const [customCategory, setCustomCategory] = useState("");

  const API_URL = "http://127.0.0.1:8000/api/offers";

  // Recherche
  const [searchTerm, setSearchTerm] = useState("");

  // Édition
  const [editingOffer, setEditingOffer] = useState<Offer | null>(null);
  // ✅ AJOUTE CES STATES POUR LA VENTE
  
  const [sellingOffer, setSellingOffer] = useState<Offer | null>(null);
  const [clients, setClients] = useState<any[]>([]);
  const [searchClient, setSearchClient] = useState("");
  const [selectedClient, setSelectedClient] = useState<number | null>(null);
  const filteredClients = clients.filter(client =>
  client.name.toLowerCase().includes(searchClient.toLowerCase()) ||
  client.company.toLowerCase().includes(searchClient.toLowerCase())
  );
  
  useEffect(() => {
    fetch(API_URL)
      .then(res => res.json())
      .then((data: Offer[]) => setOffers(data))
      .catch(err => console.error("Erreur API", err));
  }, []);

  //ajout useEffect 
  // Ajoute APRÈS tes autres useEffects
  useEffect(() => {
    if (sellingOffer) {
      fetch("http://127.0.0.1:8000/api/clients")
        .then(res => res.json())
        .then(data => setClients(data))
        .catch(err => console.error("Erreur chargement clients", err));
    }
  }, [sellingOffer]);

  // Liste filtrée par recherche
  const filteredOffers = offers
    .filter(offer =>
      offer.title.toLowerCase().includes(searchTerm.toLowerCase()) ||
      offer.description.toLowerCase().includes(searchTerm.toLowerCase())
    )
    .filter(offer => !id || offer.id.toString() === id);

  // ✅ Calculs dynamiques pour les statistiques
  const activeOffersCount = offers.filter(o => o.status === "active").length;
  const totalSales = offers.reduce((sum, o) => sum + o.clientsCount, 0);
  const totalRevenue = offers.reduce((sum, o) => {
    const price = parseFloat(o.price.replace(/[^0-9.-]+/g, ""));
    return sum + (price * o.clientsCount);
  }, 0);

  const handleEditOffer = (offer: Offer) => {
    setEditingOffer(offer);
    
    // Vérifier si c'est une catégorie personnalisée (pas dans la liste prédéfinie)
    const predefinedCategories = ["web", "seo", "marketing", "design", "consulting"];
    const isCustom = !predefinedCategories.includes(offer.category);
    
    setIsCustomCategory(isCustom);
    setCustomCategory(isCustom ? offer.category : "");
    
    setNewOffer({
      title: offer.title,
      description: offer.description,
      price: offer.price,
      currency: offer.currency || "EUR",
      category: offer.category,
      validUntil: offer.validUntil || "",
      delivery_date: offer.delivery_date || "",
      avatar: offer.avatar || "",
      images: offer.images || [], // Charger les images existantes
      condition_state: offer.condition_state || "new",
      tags: offer.tags || [],
      reviews: offer.reviews || [],
    });
    setIsDialogOpen(true);
  };

  const handleSellOffer = (offerId: number, title: string) => {
    // Trouver l'offre à vendre
    const offer = offers.find(o => o.id === offerId);
    if (!offer) return;

    // Requête PATCH/PUT vers l'API pour incrémenter clientsCount
    fetch(`${API_URL}/${offerId}`, {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        ...offer,
        clientsCount: offer.clientsCount + 1
      })
    })
      .then(res => res.json())
      .then((updatedOffer: Offer) => {
        setOffers(prev => prev.map(o => o.id === offerId ? updatedOffer : o));
        toast({
          title: "Vente enregistrée",
          description: `Un client ajouté à "${updatedOffer.title}".`,
        });
      })
      .catch(err => {
        toast({
          title: "Erreur",
          description: "Impossible d'enregistrer la vente.",
          variant: "destructive"
        });
        console.error(err);
      });
  };

  //suppression
  const handleDeleteOffer = (offerId: number, offerTitle: string) => {
    const confirmed = window.confirm(`Voulez-vous vraiment supprimer l'offre "${offerTitle}" ?`);
    if (!confirmed) return;

    fetch(`${API_URL}/${offerId}`, { method: "DELETE" })
      .then(() => {
        setOffers(prev => prev.filter(o => o.id !== offerId));
        toast({ title: "Offre supprimée", description: `L'offre "${offerTitle}" a été retirée.` });
      })
      .catch(err => {
        toast({ title: "Erreur", description: "Impossible de supprimer l'offre.", variant: "destructive" });
        console.error(err);
      });

  };

  const handleCreateOrUpdateOffer = () => {
    if (!newOffer.title || !newOffer.price || !newOffer.category) {
      toast({
        title: "Erreur",
        description: "Veuillez remplir tous les champs obligatoires.",
        variant: "destructive",
      });
      return;
    }

    const method = editingOffer ? "PUT" : "POST";
    const url = editingOffer ? `${API_URL}/${editingOffer.id}` : API_URL;

    fetch(url, {
      method,
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        ...newOffer,
        status: editingOffer ? editingOffer.status : "active",
        clientsCount: editingOffer ? editingOffer.clientsCount : 0
      }),
    })
      .then(res => res.json())
      .then((savedOffer: Offer) => {
        if (editingOffer) {
          setOffers(prev => prev.map(o => o.id === savedOffer.id ? savedOffer : o));
          toast({ title: "Offre modifiée", description: `"${savedOffer.title}" a été mise à jour.` });
        } else {
          setOffers(prev => [...prev, savedOffer]);
          toast({ title: "Offre créée", description: `"${savedOffer.title}" a été ajoutée avec succès.` });
        }
        setNewOffer({ 
          title: "", 
          description: "", 
          price: "", 
          currency: "EUR",
          category: "", 
          validUntil: "",
          delivery_date: "",
          avatar: "",
          images: [], // Reset des images
          condition_state: "new",
          tags: [],
          reviews: [],
        });
        setEditingOffer(null);
        setIsCustomCategory(false);
        setCustomCategory("");
        setIsDialogOpen(false);
      })
      .catch(err => {
        console.error("Erreur API", err);
        toast({ title: "Erreur", description: "Impossible de sauvegarder l'offre.", variant: "destructive" });
      });
  };

  return (
  <>
    <AgencyLayout>
      <div className="space-y-6">
        {/* Header */}
        <div className="flex items-center justify-between">
          <div>
            <h1 className="text-2xl font-bold text-foreground">{t("title")}</h1>
            <p className="text-muted-foreground">{t("subtitle")}</p>
          </div>
          <Dialog open={isDialogOpen} onOpenChange={setIsDialogOpen}>
            <DialogTrigger asChild>
              <Button className="bg-gradient-primary text-primary-foreground hover:opacity-90">
                <Plus className="w-4 h-4 mr-2" />
                {t("addOffer")}
              </Button>
            </DialogTrigger>
            <DialogContent className="sm:max-w-[600px]">
              <DialogHeader>
                <DialogTitle>
                  {/* //modifier */}
                {t(editingOffer ? "editOffer" : "addOffer")}
                </DialogTitle>
                <DialogDescription>
                  {/* modifier */}
                  {t(editingOffer ? "editDesc" : "addDesc")}
                </DialogDescription>
              </DialogHeader>
              <div className="grid gap-4 py-4 max-h-[60vh] overflow-y-auto">
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="title" className="text-right">
                    {/* modifier */}
                    {t("title")} *
                  </Label>
                  <Input
                    id="title"
                    value={newOffer.title}
                    onChange={(e) => setNewOffer((prev) => ({ ...prev, title: e.target.value }))}
                    className="col-span-3"
                    placeholder="Pack Site Web Premium"
                  />
                </div>
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="description" className="text-right">
                    {/* modifier */}
                    {t("description")}
                  </Label>
                  <Textarea
                    id="description"
                    value={newOffer.description}
                    onChange={(e) => setNewOffer((prev) => ({ ...prev, description: e.target.value }))}
                    className="col-span-3"
                    placeholder={t("descriptionPlaceholder")}
                    rows={3}
                  />
                </div>
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="price" className="text-right">
                    {t("price")} *
                  </Label>
                  <div className="col-span-3 flex gap-2">
                    <Input
                      id="price"
                      type="number"
                      value={newOffer.price}
                      onChange={(e) => setNewOffer((prev) => ({ ...prev, price: e.target.value }))}
                      className="flex-1"
                      placeholder="299"
                    />
                    <Select
                      value={newOffer.currency}
                      onValueChange={(value) => setNewOffer((prev) => ({ ...prev, currency: value }))}
                    >
                      <SelectTrigger className="w-24">
                        <SelectValue />
                      </SelectTrigger>
                      <SelectContent>
                        <SelectItem value="EUR">EUR</SelectItem>
                        <SelectItem value="USD">USD</SelectItem>
                        <SelectItem value="GBP">GBP</SelectItem>
                        <SelectItem value="CAD">CAD</SelectItem>
                        <SelectItem value="MGA">MGA</SelectItem>
                        <SelectItem value="XOF">XOF</SelectItem>
                      </SelectContent>
                    </Select>
                  </div>
                </div>
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="category" className="text-right">
                    {/* modifier */}
                    {t("category", { ns: "common" })} *
                  </Label>
                  <Select
                    value={isCustomCategory ? "custom" : newOffer.category}
                    onValueChange={(value) => {
                      if (value === "custom") {
                        setIsCustomCategory(true);
                        setCustomCategory("");
                        setNewOffer((prev) => ({ ...prev, category: "" }));
                      } else {
                        setIsCustomCategory(false);
                        setCustomCategory("");
                        setNewOffer((prev) => ({ ...prev, category: value }));
                      }
                    }}
                  >
                    <SelectTrigger className="col-span-3">
                    <SelectValue placeholder={t("category", { ns: "common" })} />
                    </SelectTrigger>
                    <SelectContent>
                    <SelectItem value="web">{t("category.web", { ns: "offers" })}</SelectItem>
                      <SelectItem value="seo">{t("category.seo")}</SelectItem>
                      <SelectItem value="marketing">{t("category.marketing")}</SelectItem>
                      <SelectItem value="design">{t("category.design")}</SelectItem>
                      <SelectItem value="consulting">{t("category.consulting")}</SelectItem>
                      <SelectItem value="custom">+ Autre</SelectItem>
                    </SelectContent>
                  </Select>
                </div>
                {/* Champ pour catégorie personnalisée */}
                {isCustomCategory && (
                  <div className="grid grid-cols-4 items-center gap-4">
                    <Label htmlFor="customCategory" className="text-right">
                      Catégorie personnalisée *
                    </Label>
                    <Input
                      id="customCategory"
                      value={customCategory}
                      onChange={(e) => {
                        setCustomCategory(e.target.value);
                        setNewOffer((prev) => ({ ...prev, category: e.target.value }));
                      }}
                      className="col-span-3"
                      placeholder="Entrez votre catégorie..."
                    />
                  </div>
                )}
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="validUntil" className="text-right">
                    {t("validUntil")}
                  </Label>
                  <Input
                    id="validUntil"
                    type="date"
                    value={newOffer.validUntil}
                    onChange={(e) => setNewOffer((prev) => ({ ...prev, validUntil: e.target.value }))}
                    className="col-span-3"
                  />
                </div>
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="delivery_date" className="text-right">
                    Date de livraison
                  </Label>
                  <Input
                    id="delivery_date"
                    type="date"
                    value={newOffer.delivery_date}
                    onChange={(e) => setNewOffer((prev) => ({ ...prev, delivery_date: e.target.value }))}
                    className="col-span-3"
                  />
                </div>
                <div className="grid grid-cols-4 items-start gap-4">
                  <Label className="text-right pt-2">
                    Images
                  </Label>
                  <div className="col-span-3">
                    <MultipleImageUpload
                      value={newOffer.images}
                      onChange={(urls) => setNewOffer((prev) => ({ 
                        ...prev, 
                        images: urls,
                        avatar: urls[0] || "" // La première image devient l'avatar
                      }))}
                      maxImages={5}
                    />
                  </div>
                </div>
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="condition_state" className="text-right">
                    État
                  </Label>
                  <Select
                    value={newOffer.condition_state}
                    onValueChange={(value) => setNewOffer((prev) => ({ ...prev, condition_state: value as any }))}
                  >
                    <SelectTrigger className="col-span-3">
                      <SelectValue />
                    </SelectTrigger>
                    <SelectContent>
                      <SelectItem value="new">Neuf</SelectItem>
                      <SelectItem value="like_new">Comme neuf</SelectItem>
                      <SelectItem value="good">Bon état</SelectItem>
                      <SelectItem value="fair">État correct</SelectItem>
                      <SelectItem value="poor">Mauvais état</SelectItem>
                      <SelectItem value="to_repair">À réparer</SelectItem>
                    </SelectContent>
                  </Select>
                </div>
                <div className="grid grid-cols-4 items-center gap-4">
                  <Label htmlFor="tags" className="text-right">
                    Tags
                  </Label>
                  <Input
                    id="tags"
                    value={Array.isArray(newOffer.tags) ? newOffer.tags.join(', ') : ''}
                    onChange={(e) => setNewOffer((prev) => ({ 
                      ...prev, 
                      tags: e.target.value.split(',').map(tag => tag.trim()).filter(Boolean)
                    }))}
                    className="col-span-3"
                    placeholder="design, ui, ux, moderne"
                  />
                </div>
              </div>
              <DialogFooter>
                <Button variant="outline" onClick={() => {
                  setIsDialogOpen(false);
                  setEditingOffer(null);
                  setNewOffer({ 
                    title: "", 
                    description: "", 
                    price: "", 
                    currency: "EUR",
                    category: "", 
                    validUntil: "",
                    delivery_date: "",
                    avatar: "",
                    images: [], // Reset des images
                    condition_state: "new",
                    tags: [],
                    reviews: [],
                  });
                  setIsCustomCategory(false);
                  setCustomCategory("");
                }}>
                  {t("cancel")}
                </Button>
                <Button onClick={handleCreateOrUpdateOffer}>
                  {t(editingOffer ? "saveChanges" : "create")}
                </Button>
              </DialogFooter>
            </DialogContent>
          </Dialog>
        </div>

        {/* ✅ Filters gardés */}
        <Card>
          <CardContent className="p-4">
            <div className="flex flex-col sm:flex-row gap-4">
              <div className="relative flex-1">
                <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-muted-foreground w-4 h-4" />
                <Input
                  type="text"
                  placeholder={t("search")}
                  value={searchTerm}
                  onChange={(e) => setSearchTerm(e.target.value)}
                  className="pl-10"
                />
              </div>
              <Select>
                <SelectTrigger className="w-full sm:w-48">
                  <Filter className="w-4 h-4 mr-2" />
                  <SelectValue placeholder={t("category", { ns: "common" })} />
                </SelectTrigger>
                <SelectContent>
                <SelectItem value="all">{t("allCategories")}</SelectItem>
                <SelectItem value="web">{t("category.web")}</SelectItem>
                <SelectItem value="seo">{t("category.seo")}</SelectItem>
                <SelectItem value="marketing">{t("category.marketing")}</SelectItem>
                </SelectContent>
              </Select>
              <Select>
                <SelectTrigger className="w-full sm:w-48">
                  <SelectValue placeholder={t("status")} />
                </SelectTrigger>
                <SelectContent>
                <SelectItem value="all">{t("allStatuses")}</SelectItem>
                <SelectItem value="active">{t("active")}</SelectItem>
                <SelectItem value="pending">{t("pending")}</SelectItem>
                </SelectContent>
              </Select>
            </div>
          </CardContent>
        </Card>

        {/* ✅ Offers Grid gardée */}
        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
          {filteredOffers.map((offer) => (
            <Card key={offer.id} className="hover:shadow-md transition-shadow">
              <CardHeader>
                <div className="flex items-start justify-between">
                  <div className="space-y-1">
                    <CardTitle className="text-lg">{offer.title}</CardTitle>
                    <p className="text-sm text-muted-foreground">{offer.description}</p>
                  </div>
                  {getStatusBadge(offer.status, t)}
                </div>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="flex items-center justify-between">
                  <span className="text-2xl font-bold text-primary">{offer.price}€</span>
                  <Badge variant="outline" className="text-xs">
                  {offer.clientsCount} {t(offer.clientsCount > 1 ? "soldPlural" : "soldSingular")}
                  </Badge>
                </div>
                {offer.validUntil && (
                  <div className="text-xs text-muted-foreground">
                    {t("validUntil")} {new Date(offer.validUntil).toLocaleDateString("fr-FR")}
                  </div>
                )}
                {/* Boutons actions */}
                <div className="flex gap-2">
                  <Button
                    variant="outline"
                    size="sm"
                    className="flex-1"
                    onClick={() => handleEditOffer(offer)}
                  >
                    <Edit className="w-4 h-4 mr-1" />
                    {t("edit")}
                  </Button>
                  <Button
                    variant="destructive"
                    size="sm"
                    className="flex-1"
                    onClick={() => setOfferToDelete(offer)}>
                    {t("delete")}
                  </Button>
                </div>
                {/* ✅ Bouton vendre sur une ligne à part */}
                <Button
                  size="sm"
                  className="w-full mt-2"
                  onClick={() => setSellingOffer(offer)}
                  //onClick={() => handleSellOffer(offer.id, offer.title)}
                >
                  <ShoppingCart className="w-4 h-4 mr-1" />
                  {t("sell")}
                </Button>

                <Link to={`/agency/offres/${offer.id}`} className="block mt-3">
                  <Button variant="secondary" size="sm" className="w-full">
                    {t("profileDetails")}
                  </Button>
                </Link>
              </CardContent>
            </Card>
          ))}
        </div>

        {/* ✅ Stats dynamiques */}
        <div className="grid grid-cols-1 md:grid-cols-3 gap-6">
          <Card>
            <CardContent className="p-6">
              <div className="flex items-center space-x-2">
                <div className="p-2 bg-primary/10 rounded-lg">
                  <ShoppingCart className="w-5 h-5 text-primary" />
                </div>
                <div>
                  <p className="text-sm text-muted-foreground">{t("activeOffers")}</p>
                  <p className="text-2xl font-bold">{activeOffersCount}</p>
                </div>
              </div>
            </CardContent>
          </Card>
          <Card>
            <CardContent className="p-6">
              <div className="flex items-center space-x-2">
                <div className="p-2 bg-success/10 rounded-lg">
                  <CheckCircle className="w-5 h-5 text-success" />
                </div>
                <div>
                  <p className="text-sm text-muted-foreground">{t("totalSales")}</p>
                  <p className="text-2xl font-bold">{totalSales}</p>
                </div>
              </div>
            </CardContent>
          </Card>
          <Card>
            <CardContent className="p-6">
              <div className="flex items-center space-x-2">
                <div className="p-2 bg-warning/10 rounded-lg">
                  <Clock className="w-5 h-5 text-warning" />
                </div>
                <div>
                  <p className="text-sm text-muted-foreground">{t("totalRevenue")}</p>
                  <p className="text-2xl font-bold">{totalRevenue.toLocaleString('fr-FR')}€</p>
                </div>
              </div>
            </CardContent>
          </Card>
        </div>
      </div>

      {/* ✅ Confirmation suppression */}
      <Dialog open={!!offerToDelete} onOpenChange={() => setOfferToDelete(null)}>
        <DialogContent className="sm:max-w-[400px]">
          <DialogHeader>
            <DialogTitle>{t("confirmDelete")}</DialogTitle>
            <DialogDescription>
              {t("confirmDeleteDesc", { title: offerToDelete?.title })}
            </DialogDescription>
          </DialogHeader>
          <DialogFooter>
            <Button variant="outline" onClick={() => setOfferToDelete(null)}>
              {t("cancel")}
            </Button>
            <Button
              variant="destructive"
              onClick={() => {
                fetch(`${API_URL}/${offerToDelete?.id}`, { method: "DELETE" })
                  .then(() => {
                    setOffers(prev => prev.filter(o => o.id !== offerToDelete?.id));
                    toast({
                      title: "Offre supprimée",
                      description: `L'offre "${offerToDelete?.title}" a été retirée.`
                    });
                    setOfferToDelete(null);
                  })
                  .catch(err => {
                    toast({
                      title: "Erreur",
                      description: "Impossible de supprimer l'offre.",
                      variant: "destructive"
                    });
                    console.error(err);
                  });
              }}
            >
              {t("delete")}
            </Button>
          </DialogFooter>
        </DialogContent>
      </Dialog>
    </AgencyLayout>
{/* MODAL DE VENTE SIMPLIFIÉ */}
<Dialog open={!!sellingOffer} onOpenChange={() => setSellingOffer(null)}>
  <DialogContent className="sm:max-w-[500px]">
    <DialogHeader>
      <DialogTitle>{t("offers.sell")} {sellingOffer?.title}</DialogTitle>
      <DialogDescription>
        {t("selectClient")}
      </DialogDescription>
    </DialogHeader>
    <div className="space-y-4">
      {/* BARRE DE RECHERCHE */}
        <Input 
          placeholder={t("common.search")} 
          value={searchClient}
          onChange={(e) => setSearchClient(e.target.value)}
        />
        <div className="border rounded-lg p-2 h-40 overflow-y-auto">
          {/* UTILISE filteredClients AU LIEU DE clients */}
          {filteredClients.map(client => (
            <div 
              key={client.id} 
              className="p-2 hover:bg-gray-100 rounded cursor-pointer"
              onClick={() => {
                fetch(`http://127.0.0.1:8000/api/offers/${sellingOffer!.id}/attach-client/${client.id}`, {
                    method: "POST",
                    headers: {
                      "Content-Type": "application/json",
                      "Authorization": `Bearer ${localStorage.getItem('token')}`
                    }
                  })
                  .then(() => {
                    setOffers(prev => prev.map(o => 
                      o.id === sellingOffer!.id 
                        ? {...o, clientsCount: o.clientsCount + 1}
                        : o
                    ));
                  })
                toast({ 
                  title: "Vente enregistrée", 
                  description: `${client.name} a acheté "${sellingOffer!.title}"` 
                });
                setSellingOffer(null);
              }}
            >
              {client.name} - {client.company}
            </div>
          ))}
        </div>
      </div>
    </DialogContent>
  </Dialog>
  </>
      
  ); 
}
