// agency-frontend/src/pages/agency/AgencySettings.tsx
import { AgencyLayout } from "@/components/agency/AgencyLayout";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import { Textarea } from "@/components/ui/textarea";
import { Switch } from "@/components/ui/switch";
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
import { Separator } from "@/components/ui/separator";
import { Badge } from "@/components/ui/badge";
import { Building2, User, Bell, Shield, CreditCard, Globe , Mail, Edit, Trash2, Crown, Users, AlertTriangle, ArrowLeft} from "lucide-react";
import { useTranslation } from "react-i18next";
import { useSettings } from "@/context/SettingsContext";
import { useState, useEffect } from "react";
import { useToast } from "@/hooks/use-toast";
import { Agency, AgencyFormData } from "@/types/agency";
import { fetchAgency, createOrUpdateAgency } from "@/lib/agency-api";
import { Collaborator } from "@/types/invitation";
import { fetchCollaborators, deleteCollaborator } from "@/lib/invitation-api";
import { InviteCollaboratorModal } from "@/components/agency/InviteCollaboratorModal";
import { EditPermissionsModal } from "@/components/agency/EditPermissionsModal";
import { useAuth } from "@/context/AuthContext";
import { useNavigate } from "react-router-dom";


export default function AgencySettings() {
  const { t } = useTranslation("settings");
  const { language, timezone, setLanguage, setTimezone } = useSettings();
  const { toast } = useToast();
  const { user: authUser, hasPermission, isAdmin, isLoading } = useAuth();
  const navigate = useNavigate();
  
  // ✅ Vérification des permissions
  const canViewSettings = isAdmin() || hasPermission('view_settings');
  const canManageAgency = isAdmin() || hasPermission('manage_agency');
  const canManageTeam = isAdmin() || hasPermission('manage_team');

  // ✅ Convertir l'utilisateur auth en Collaborator
  const currentUser = authUser ? {
    id: authUser.id,
    name: authUser.name,
    email: authUser.email,
    role: (authUser as any).role || 'user',
    avatar: authUser.avatar,
    permissions: (authUser as any).permissions || [],
    invited_at: null,
    accepted_at: null,
    status: 'active',
    inviter_name: null
  } as Collaborator : null;

  const [agency, setAgency] = useState<Agency | null>(null);
  const [loading, setLoading] = useState(false);
  const [formData, setFormData] = useState<AgencyFormData>({
    name: "",
    email: "",
    phone: "",
    website: "",
    address: ""
  });

  // États pour la gestion des utilisateurs
  const [collaborators, setCollaborators] = useState<Collaborator[]>([]);
  const [inviteModalOpen, setInviteModalOpen] = useState(false);
  const [editPermissionsModalOpen, setEditPermissionsModalOpen] = useState(false);
  const [selectedCollaborator, setSelectedCollaborator] = useState<Collaborator | null>(null);
  const [loadingCollaborators, setLoadingCollaborators] = useState(false);

  const loadAgency = async () => {
    try {
      const agencyData = await fetchAgency();
      setAgency(agencyData);
      setFormData({
        name: agencyData.name,
        email: agencyData.email,
        phone: agencyData.phone || "",
        website: agencyData.website || "",
        address: agencyData.address || ""
      });
    } catch (error) {
      console.log("Aucune agence existante, formulaire vierge affiché");
    }
  };

  const loadCollaborators = async () => {
    setLoadingCollaborators(true);
    try {
      const data = await fetchCollaborators();
      setCollaborators(data);
    } catch (error) {
      toast({
        title: "Erreur",
        description: "Erreur lors du chargement des utilisateurs",
        variant: "destructive"
      });
    } finally {
      setLoadingCollaborators(false);
    }
  };

  useEffect(() => {
    if (!canViewSettings) return;
    loadAgency();
    if (canManageTeam) {
      loadCollaborators();
    }
  }, [canViewSettings, canManageTeam]);

  // ✅ Pendant le chargement de l'authentification, on affiche un état de chargement
  if (isLoading) {
    return (
      <AgencyLayout>
        <div className="flex items-center justify-center min-h-[60vh]">
          <div className="text-center text-muted-foreground">Chargement des permissions...</div>
        </div>
      </AgencyLayout>
    );
  }

  // ✅ SI L'UTILISATEUR N'A PAS LA PERMISSION
  if (!canViewSettings) {
    return (
      <AgencyLayout>
        <div className="flex items-center justify-center min-h-[60vh]">
          <div className="text-center max-w-md">
            <AlertTriangle className="h-16 w-16 text-yellow-500 mx-auto mb-4" />
            <h2 className="text-2xl font-bold text-foreground mb-2">Accès restreint</h2>
            <p className="text-muted-foreground mb-6">
              Vous n'avez pas la permission d'accéder aux paramètres.
            </p>
            <div className="flex gap-3 justify-center">
              <Button variant="outline" onClick={() => navigate(-1)}>
                <ArrowLeft className="w-4 h-4 mr-2" />
                Retour
              </Button>
              <Button onClick={() => navigate("/agency")}>
                Accueil
              </Button>
            </div>
          </div>
        </div>
      </AgencyLayout>
    );
  }

  const handleInputChange = (field: keyof AgencyFormData, value: string) => {
    setFormData(prev => ({
      ...prev,
      [field]: value
    }));
  };

  const handleSaveAgency = async () => {
    if (!canManageAgency) {
      toast({
        title: "Permission refusée",
        description: "Vous n'avez pas la permission de modifier les paramètres de l'agence.",
        variant: "destructive"
      });
      return;
    }

    if (!formData.name || !formData.email) {
      toast({
        title: "Erreur",
        description: "Le nom et l'email sont obligatoires",
        variant: "destructive"
      });
      return;
    }

    setLoading(true);
    try {
      const result = await createOrUpdateAgency(formData);
      setAgency(result.agency);
      toast({
        title: "Succès",
        description: result.message
      });
    } catch (error) {
      toast({
        title: "Erreur",
        description: error instanceof Error ? error.message : "Erreur inconnue",
        variant: "destructive"
      });
    } finally {
      setLoading(false);
    }
  };

  const handleEditPermissions = (collaborator: Collaborator) => {
    setSelectedCollaborator(collaborator);
    setEditPermissionsModalOpen(true);
  };

  const getDisplayName = (user: Collaborator) => {
    if (user.name) {
      const cleanName = user.name.replace(/^Collaborateur\s+/, '');
      return cleanName || user.email;
    }
    return user.email;
  };

  const handleDeleteCollaborator = async (collaborator: Collaborator) => {
    if (!canManageTeam) {
      toast({
        title: "Permission refusée",
        description: "Vous n'avez pas la permission de supprimer des collaborateurs.",
        variant: "destructive"
      });
      return;
    }

    const displayName = getDisplayName(collaborator);
    if (!confirm(`Êtes-vous sûr de vouloir supprimer ${displayName} ? Cette action est irréversible.`)) {
      return;
    }

    try {
      await deleteCollaborator(collaborator.id);
      toast({
        title: "Succès",
        description: "Collaborateur supprimé avec succès"
      });
      loadCollaborators();
    } catch (error) {
      toast({
        title: "Erreur",
        description: error instanceof Error ? error.message : "Erreur lors de la suppression",
        variant: "destructive"
      });
    }
  };

  const getStatusBadge = (user: Collaborator) => {
    if (user.role === 'admin') {
      return (
        <Badge className="bg-purple-100 text-purple-800 border-purple-200">
          <Crown className="w-3 h-3 mr-1" />
          Administrateur
        </Badge>
      );
    }
    
    if (user.status === 'pending') {
      return <Badge variant="secondary" className="bg-yellow-100 text-yellow-800">En attente</Badge>;
    }
    
    return <Badge variant="secondary">Collaborateur</Badge>;
  };

  const getPermissionText = (permissions: string[] | null) => {
    if (!permissions || permissions.length === 0) return "Aucune permission";
    return `${permissions.length} permission${permissions.length > 1 ? 's' : ''}`;
  };

  const isCurrentUser = (user: Collaborator) => {
    return user.id === currentUser?.id;
  };

  return (
    <AgencyLayout>
      <div className="space-y-6">
        <div>
          <h1 className="text-2xl font-bold text-foreground">{t("title")}</h1>
          <p className="text-muted-foreground">{t("description")}</p>
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-3 gap-6">
          <div className="lg:col-span-2 space-y-6">
            <Card>
              <CardHeader>
                <CardTitle className="flex items-center gap-2">
                  <Building2 className="w-5 h-5" /> {t("agencyInfo")}
                </CardTitle>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                  <div className="space-y-2">
                    <Label htmlFor="agency-name">Nom de l'agence *</Label>
                    <Input 
                      id="agency-name" 
                      value={formData.name}
                      onChange={(e) => handleInputChange('name', e.target.value)}
                      placeholder="Nom de votre agence" 
                      disabled={!canManageAgency}
                    />
                  </div>
                  <div className="space-y-2">
                    <Label htmlFor="agency-email">Email principal *</Label>
                    <Input 
                      id="agency-email" 
                      type="email" 
                      value={formData.email}
                      onChange={(e) => handleInputChange('email', e.target.value)}
                      placeholder="email@votre-agence.fr" 
                      disabled={!canManageAgency}
                    />
                  </div>
                </div>
                <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                  <div className="space-y-2">
                    <Label htmlFor="agency-phone">Téléphone</Label>
                    <Input 
                      id="agency-phone" 
                      value={formData.phone}
                      onChange={(e) => handleInputChange('phone', e.target.value)}
                      placeholder="+33 1 23 45 67 89" 
                      disabled={!canManageAgency}
                    />
                  </div>
                  <div className="space-y-2">
                    <Label htmlFor="agency-website">Site web</Label>
                    <Input 
                      id="agency-website" 
                      value={formData.website}
                      onChange={(e) => handleInputChange('website', e.target.value)}
                      placeholder="https://votre-agence.fr" 
                      disabled={!canManageAgency}
                    />
                  </div>
                </div>
                <div className="space-y-2">
                  <Label htmlFor="agency-address">Adresse</Label>
                  <Textarea 
                    id="agency-address" 
                    value={formData.address}
                    onChange={(e) => handleInputChange('address', e.target.value)}
                    placeholder="123 Rue de la République&#10;75001 Paris, France" 
                    rows={3} 
                    disabled={!canManageAgency}
                  />
                </div>

                {canManageAgency && (
                  <Button 
                    onClick={handleSaveAgency}
                    disabled={loading}
                  >
                    {loading ? "Enregistrement..." : "Enregistrer les modifications"}
                  </Button>
                )}
                
                {!canManageAgency && (
                  <div className="text-sm text-muted-foreground p-3 bg-yellow-50 rounded-lg">
                    <AlertTriangle className="w-4 h-4 inline mr-2" />
                    Vous n'avez pas la permission de modifier les paramètres de l'agence.
                  </div>
                )}
                
                {agency && (
                  <div className="text-sm text-muted-foreground">
                    Dernière modification: {new Date(agency.updated_at).toLocaleDateString('fr-FR')}
                  </div>
                )}
              </CardContent>
            </Card>

            {/* User Management - Seulement si permission manage_team */}
            {canManageTeam && (
              <Card>
                <CardHeader>
                  <CardTitle className="flex items-center gap-2">
                    <Users className="w-5 h-5" />
                    Gestion des utilisateurs
                  </CardTitle>
                </CardHeader>
                <CardContent className="space-y-4">
                  {loadingCollaborators ? (
                    <div className="text-center py-4">
                      <p>Chargement des utilisateurs...</p>
                    </div>
                  ) : collaborators.length === 0 ? (
                    <div className="text-center py-8 text-muted-foreground">
                      <Users className="w-12 h-12 mx-auto mb-2 opacity-50" />
                      <p>Aucun utilisateur pour le moment</p>
                      <p className="text-sm">Invitez des collaborateurs pour travailler avec vous</p>
                    </div>
                  ) : (
                    <>
                      {collaborators.map((user) => (
                        <div key={user.id} className="flex items-center justify-between p-4 border rounded-lg">
                          <div className="flex items-center gap-3">
                            <div className="w-10 h-10 bg-primary/10 rounded-full flex items-center justify-center">
                              {user.avatar ? (
                                <img 
                                  src={user.avatar} 
                                  alt={getDisplayName(user)}
                                  className="w-10 h-10 rounded-full"
                                />
                              ) : (
                                <User className="w-5 h-5 text-primary" />
                              )}
                            </div>
                            <div className="flex-1">
                              <div className="flex items-center gap-2">
                                <p className="font-medium">
                                  {getDisplayName(user)}
                                  {isCurrentUser(user) && (
                                    <span className="text-sm text-muted-foreground ml-2">(Vous)</span>
                                  )}
                                </p>
                                {getStatusBadge(user)}
                              </div>
                              <p className="text-sm text-muted-foreground">
                                {user.email}
                              </p>
                              {user.role === 'collaborator' && (
                                <p className="text-xs text-muted-foreground">
                                  {getPermissionText(user.permissions)}
                                  {user.status === 'pending' && ' • Invitation en attente'}
                                  {user.inviter_name && ` • Invité par ${user.inviter_name}`}
                                </p>
                              )}
                            </div>
                          </div>
                          <div className="flex items-center gap-2">
                            {user.role === 'collaborator' && (
                              <>
                                <Button 
                                  variant="outline" 
                                  size="sm"
                                  onClick={() => handleEditPermissions(user)}
                                >
                                  <Edit className="w-4 h-4 mr-1" />
                                  Permissions
                                </Button>
                                <Button 
                                  variant="outline" 
                                  size="sm"
                                  onClick={() => handleDeleteCollaborator(user)}
                                  className="text-red-600 hover:text-red-700 hover:bg-red-50 border-red-200"
                                >
                                  <Trash2 className="w-4 h-4" />
                                </Button>
                              </>
                            )}
                            {user.role === 'admin' && (
                              <Badge variant="outline" className="text-green-600 border-green-200">
                                Propriétaire
                              </Badge>
                            )}
                          </div>
                        </div>
                      ))}
                    </>
                  )}

                  <Button 
                    variant="outline" 
                    className="w-full"
                    onClick={() => setInviteModalOpen(true)}
                  >
                    <Mail className="w-4 h-4 mr-2" />
                    Inviter un collaborateur
                  </Button>
                </CardContent>
              </Card>
            )}

            {/* Notifications */}
            <Card>
              <CardHeader>
                <CardTitle className="flex items-center gap-2">
                  <Bell className="w-5 h-5" /> {t("notifications")}
                </CardTitle>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="flex items-center justify-between">
                  <div>
                    <p className="font-medium">{t("newClients")}</p>
                    <p className="text-sm text-muted-foreground">{t("newClientsDesc")}</p>
                  </div>
                  <Switch defaultChecked />
                </div>
                <Separator />
                <div className="flex items-center justify-between">
                  <div>
                    <p className="font-medium">{t("orders")}</p>
                    <p className="text-sm text-muted-foreground">{t("ordersDesc")}</p>
                  </div>
                  <Switch defaultChecked />
                </div>
              </CardContent>
            </Card>
          </div>

          <div className="space-y-6">
            <Card>
              <CardHeader>
                <CardTitle className="flex items-center gap-2">
                  <Shield className="w-5 h-5" /> {t("security")}
                </CardTitle>
              </CardHeader>
              <CardContent className="space-y-4">
                <Button variant="outline" className="w-full justify-start">{t("changePassword")}</Button>
                <Button variant="outline" className="w-full justify-start">{t("twoFactor")}</Button>
              </CardContent>
            </Card>

            <Card>
              <CardHeader>
                <CardTitle className="flex items-center gap-2">
                  <CreditCard className="w-5 h-5" /> {t("billing")}
                </CardTitle>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="p-3 bg-primary/5 rounded-lg">
                  <p className="font-medium">Plan Pro</p>
                  <p className="text-sm text-muted-foreground">99€/mois</p>
                </div>
                <Button variant="outline" className="w-full justify-start">{t("manageSubscription")}</Button>
              </CardContent>
            </Card>

            <Card>
              <CardHeader>
                <CardTitle className="flex items-center gap-2">
                  <Globe className="w-5 h-5" /> {t("preferences")}
                </CardTitle>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="space-y-2">
                  <Label>{t("language")}</Label>
                  <Select value={language} onValueChange={setLanguage}>
                    <SelectTrigger><SelectValue /></SelectTrigger>
                    <SelectContent>
                      <SelectItem value="fr">Français</SelectItem>
                      <SelectItem value="en">English</SelectItem>
                    </SelectContent>
                  </Select>
                </div>
                <div className="space-y-2">
                  <Label>{t("timezone")}</Label>
                  <Select value={timezone} onValueChange={setTimezone}>
                    <SelectTrigger><SelectValue /></SelectTrigger>
                    <SelectContent>
                      <SelectItem value="Europe/Paris">Europe/Paris</SelectItem>
                      <SelectItem value="Europe/London">Europe/London</SelectItem>
                      <SelectItem value="America/New_York">America/New_York</SelectItem>
                    </SelectContent>
                  </Select>
                </div>
              </CardContent>
            </Card>
          </div>
        </div>
      </div>

      {canManageTeam && (
        <>
          <InviteCollaboratorModal
            open={inviteModalOpen}
            onOpenChange={setInviteModalOpen}
            onInviteSent={loadCollaborators}
          />

          <EditPermissionsModal
            open={editPermissionsModalOpen}
            onOpenChange={setEditPermissionsModalOpen}
            collaborator={selectedCollaborator}
            onPermissionsUpdated={loadCollaborators}
          />
        </>
      )}
    </AgencyLayout>
  );
}
