// agency-frontend/src/pages/agency/OfferProfil.tsx
import { useState, useEffect } from "react";
import { useParams, Link } from "react-router-dom";
import { AgencyLayout } from "@/components/agency/AgencyLayout";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { Badge } from "@/components/ui/badge";
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import { Textarea } from "@/components/ui/textarea";
import {
  ArrowLeft,
  Edit,
  Save,
  X,
  FileText,
  CreditCard,
  Clock,
  User,
  Tag,
  Calendar,
  DollarSign,
  Users,
  Eye
} from "lucide-react";
import { useToast } from "@/hooks/use-toast";
import axios from "axios";
import { Client } from "@/types/clients";
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
} from "@/components/ui/dialog";
import { useTranslation } from "react-i18next";
import { ImageUpload } from "@/components/ui/image-upload";

interface Offer {
  id: number;
  title: string;
  description?: string;
  price?: number;
  currency?: string;
  category?: string;
  status?: "active" | "inactive" | "pending";
  validUntil?: string;
  delivery_date?: string;
  avatar?: string;
  seller_profile?: string;
  condition_state?: "new" | "like_new" | "good" | "fair" | "poor" | "to_repair";
  tags?: string[];
  reviews?: Array<{
    rating: number;
    comment: string;
    user_name?: string;
  }>;
  clients?: Client[];
  clientsCount?: number;
}

const API_URL = "http://127.0.0.1:8000/api/offers";

export default function OfferProfile() {
  const { t } = useTranslation("OfferProfil");
  const { id } = useParams<{ id: string }>();
  const { toast } = useToast();

  const [offer, setOffer] = useState<Offer | null>(null);
  const [isEditing, setIsEditing] = useState(false);
  const [formData, setFormData] = useState<Partial<Offer>>({});
  const [clientToRemove, setClientToRemove] = useState<{client: Client, offer: Offer} | null>(null);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const fetchOffer = async () => {
      try {
        setIsLoading(true);
        const res = await axios.get<Offer>(`${API_URL}/${id}`);
        setOffer(res.data);
        setFormData(res.data);
      } catch (error) {
        console.error("Erreur lors du chargement de l'offre", error);
        toast({
          title: t("error", { ns: "common" }),
          description: t("offers.profileLoadError"),
          variant: "destructive",
        });
      } finally {
        setIsLoading(false);
      }
    };

    if (id) fetchOffer();
  }, [id, toast, t]);

  const handleSave = async () => {
    try {
      if (!id) return;
      
      const res = await axios.put<Offer>(`${API_URL}/${id}`, formData);
      setOffer(res.data);
      setFormData(res.data);
      setIsEditing(false);
      toast({
        title: t("offers.profileUpdated"),
        description: t("offers.profileUpdatedDesc"),
      });
    } catch (error) {
      console.error("Erreur lors de la sauvegarde", error);
      toast({
        title: t("error", { ns: "common" }),
        description: t("offers.profileSaveError"),
        variant: "destructive",
      });
    }
  };

  const handleCancel = () => {
    if (offer) setFormData(offer);
    setIsEditing(false);
  };

  const formatDateForInput = (dateString: string | undefined) => {
    if (!dateString) return "";
    try {
      return new Date(dateString).toISOString().split('T')[0];
    } catch {
      return "";
    }
  };

  const formatDateForDisplay = (dateString: string | undefined) => {
    if (!dateString) return t("offers.profileUnlimited");
    try {
      return new Date(dateString).toLocaleDateString('fr-FR');
    } catch {
      return t("invalidDate", { ns: "common" });
    }
  };

  const getStatusBadge = (status: string) => {
    switch (status) {
      case "active":
        return <Badge className="bg-green-100 text-green-800 border-green-200">{t("active", { ns: "common" })}</Badge>;
      case "inactive":
        return <Badge className="bg-red-100 text-red-800 border-red-200">{t("inactive", { ns: "common" })}</Badge>;
      case "pending":
        return <Badge className="bg-yellow-100 text-yellow-800 border-yellow-200">{t("pending", { ns: "common" })}</Badge>;
      default:
        return <Badge variant="secondary">{t("unknown", { ns: "common" })}</Badge>;
    }
  };

  if (isLoading) {
    return (
      <AgencyLayout>
        <div className="flex items-center justify-center min-h-[400px]">
          <div className="text-center space-y-4">
            <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-primary mx-auto"></div>
            <p className="text-muted-foreground">{t("loading", { ns: "common" })}...</p>
          </div>
        </div>
      </AgencyLayout>
    );
  }

  if (!offer) {
    return (
      <AgencyLayout>
        <div className="text-center space-y-4 py-12">
          <FileText className="h-16 w-16 text-muted-foreground mx-auto" />
          <h2 className="text-2xl font-bold text-foreground">{t("offers.profileNotFound")}</h2>
          <p className="text-muted-foreground">{t("offers.profileNotFoundDesc")}</p>
          <Button asChild>
            <Link to="/agency/offers">
              <ArrowLeft className="mr-2 h-4 w-4" />
              {t("offers.profileBack")}
            </Link>
          </Button>
        </div>
      </AgencyLayout>
    );
  }

  return (
    <>
    <AgencyLayout>
      <div className="space-y-6">
        <div className="flex flex-col lg:flex-row lg:items-center lg:justify-between gap-4">
          <div className="flex items-center gap-4">
            <Button variant="outline" size="sm" asChild className="shadow-sm">
              <Link to="/agency/offers">
                <ArrowLeft className="mr-2 h-4 w-4" />
                {t("offers.profileBack")}
              </Link>
            </Button>
            <div className="space-y-1">
              <h1 className="text-2xl font-bold text-foreground flex items-center gap-2">
                <FileText className="h-6 w-6 text-primary" />
                {offer.title}
              </h1>
              <div className="flex items-center gap-3">
                {getStatusBadge(offer.status || "")}
                <Badge variant="outline" className="flex items-center gap-1">
                  <Tag className="h-3 w-3" />
                  {offer.category || t("offers.profileUncategorized")}
                </Badge>
              </div>
            </div>
          </div>

          <div className="flex items-center gap-3">
            {isEditing ? (
              <div className="flex gap-2">
                <Button size="sm" onClick={handleSave} className="bg-green-600 hover:bg-green-700 shadow-sm">
                  <Save className="mr-2 h-4 w-4" />
                  {t("save", { ns: "common" })}
                </Button>
                <Button size="sm" variant="outline" onClick={handleCancel} className="shadow-sm">
                  <X className="mr-2 h-4 w-4" />
                  {t("cancel", { ns: "common" })}
                </Button>
              </div>
            ) : (
              <Button 
                size="sm" 
                onClick={() => setIsEditing(true)} 
                className="bg-primary hover:bg-primary/90 shadow-sm"
              >
                <Edit className="mr-2 h-4 w-4" />
                {t("offers.profileEditOffer")}
              </Button>
            )}
          </div>
        </div>

        <div className="grid grid-cols-1 md:grid-cols-3 gap-4">
          <Card className="bg-gradient-to-br from-blue-50 to-blue-100 border-blue-200">
            <CardContent className="p-4 flex items-center gap-4">
              <div className="p-3 bg-blue-100 rounded-lg">
                <DollarSign className="h-6 w-6 text-blue-600" />
              </div>
              <div>
                <p className="text-sm text-blue-600 font-medium">{t("price", { ns: "common" })}</p>
                <p className="text-2xl font-bold text-blue-900">
                  {offer.price ? `${offer.price} ${offer.currency || 'EUR'}` : "—"}
                </p>
              </div>
            </CardContent>
          </Card>

          <Card className="bg-gradient-to-br from-green-50 to-green-100 border-green-200">
            <CardContent className="p-4 flex items-center gap-4">
              <div className="p-3 bg-green-100 rounded-lg">
                <Users className="h-6 w-6 text-green-600" />
              </div>
              <div>
                <p className="text-sm text-green-600 font-medium">{t("offers.clientsCount")}</p>
                <p className="text-2xl font-bold text-green-900">
                  {offer.clientsCount || 0}
                </p>
              </div>
            </CardContent>
          </Card>

          <Card className="bg-gradient-to-br from-orange-50 to-orange-100 border-orange-200">
            <CardContent className="p-4 flex items-center gap-4">
              <div className="p-3 bg-orange-100 rounded-lg">
                <Calendar className="h-6 w-6 text-orange-600" />
              </div>
              <div>
                <p className="text-sm text-orange-600 font-medium">{t("offers.validUntil")}</p>
                <p className="text-lg font-bold text-orange-900">
                  {formatDateForDisplay(offer.validUntil)}
                </p>
              </div>
            </CardContent>
          </Card>
        </div>

        <Tabs defaultValue="overview" className="space-y-6">
          <TabsList className="grid w-full grid-cols-2 lg:w-auto lg:inline-flex">
            <TabsTrigger value="overview" className="flex items-center gap-2">
              <Eye className="h-4 w-4" />
              {t("offers.profileOverview")}
            </TabsTrigger>
            <TabsTrigger value="clients" className="flex items-center gap-2">
              <User className="h-4 w-4" />
              {t("offers.profileClientsTab")}
              {offer.clients && offer.clients.length > 0 && (
                <Badge variant="secondary" className="ml-1 h-5 w-5 p-0 flex items-center justify-center text-xs">
                  {offer.clients.length}
                </Badge>
              )}
            </TabsTrigger>
          </TabsList>

          <TabsContent value="overview" className="space-y-6">
            <Card>
              <CardHeader className="bg-muted/50 border-b">
                <CardTitle className="flex items-center gap-2 text-lg">
                  <FileText className="h-5 w-5 text-primary" />
                  {t("offers.profileDetails")}
                </CardTitle>
              </CardHeader>
              <CardContent className="p-6">
                {isEditing ? (
                  <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
                    <div className="space-y-4">
                      <div>
                        <Label htmlFor="title" className="text-sm font-medium">{t("title", { ns: "common" })}</Label>
                        <Input
                          id="title"
                          value={formData.title || ""}
                          onChange={(e) => setFormData({ ...formData, title: e.target.value })}
                          className="mt-1"
                        />
                      </div>
                      <div className="grid grid-cols-2 gap-2">
                        <div>
                          <Label htmlFor="price" className="text-sm font-medium">{t("price", { ns: "common" })}</Label>
                          <Input
                            id="price"
                            type="number"
                            value={formData.price || ""}
                            onChange={(e) => setFormData({ ...formData, price: Number(e.target.value) })}
                            className="mt-1"
                          />
                        </div>
                        <div>
                          <Label htmlFor="currency" className="text-sm font-medium">Devise</Label>
                          <select
                            id="currency"
                            value={formData.currency || "EUR"}
                            onChange={(e) => setFormData({ ...formData, currency: e.target.value })}
                            className="mt-1 flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                          >
                            <option value="EUR">EUR</option>
                            <option value="USD">USD</option>
                            <option value="GBP">GBP</option>
                            <option value="CAD">CAD</option>
                            <option value="MGA">MGA</option>
                            <option value="XOF">XOF</option>
                          </select>
                        </div>
                      </div>
                      <div>
                        <Label htmlFor="category" className="text-sm font-medium">{t("category", { ns: "common" })}</Label>
                        <select
                          id="category"
                          value={formData.category || ""}
                          onChange={(e) => setFormData({ ...formData, category: e.target.value })}
                          className="mt-1 flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                        >
                          <option value="">Sélectionner une catégorie</option>
                          <option value="web">Web</option>
                          <option value="seo">SEO</option>
                          <option value="marketing">Marketing</option>
                          <option value="design">Design</option>
                          <option value="consulting">Consulting</option>
                          <option value="electronics">Électronique</option>
                          <option value="fashion">Mode</option>
                          <option value="services">Services</option>
                        </select>
                      </div>
                      <div>
                        <Label className="text-sm font-medium">Image</Label>
                        <div className="mt-1">
                          <ImageUpload
                            value={formData.avatar || ""}
                            onChange={(url) => setFormData({ ...formData, avatar: url })}
                          />
                        </div>
                      </div>
                    </div>
                    <div className="space-y-4">
                      <div>
                        <Label htmlFor="validUntil" className="text-sm font-medium">{t("offers.validUntil")}</Label>
                        <Input
                          id="validUntil"
                          type="date"
                          value={formatDateForInput(formData.validUntil)}
                          onChange={(e) => setFormData({ 
                            ...formData, 
                            validUntil: e.target.value || undefined 
                          })}
                          className="mt-1"
                        />
                      </div>
                      <div>
                        <Label htmlFor="delivery_date" className="text-sm font-medium">Date de livraison</Label>
                        <Input
                          id="delivery_date"
                          type="date"
                          value={formatDateForInput(formData.delivery_date)}
                          onChange={(e) => setFormData({ 
                            ...formData, 
                            delivery_date: e.target.value || undefined 
                          })}
                          className="mt-1"
                        />
                      </div>
                      <div>
                        <Label htmlFor="condition_state" className="text-sm font-medium">État</Label>
                        <select
                          id="condition_state"
                          value={formData.condition_state || "new"}
                          onChange={(e) => setFormData({ ...formData, condition_state: e.target.value as any })}
                          className="mt-1 flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                        >
                          <option value="new">Neuf</option>
                          <option value="like_new">Comme neuf</option>
                          <option value="good">Bon état</option>
                          <option value="fair">État correct</option>
                          <option value="poor">Mauvais état</option>
                          <option value="to_repair">À réparer</option>
                        </select>
                      </div>
                    </div>
                    <div className="lg:col-span-2 space-y-4">
                      <div>
                        <Label htmlFor="description" className="text-sm font-medium">{t("description", { ns: "common" })}</Label>
                        <Textarea
                          id="description"
                          value={formData.description || ""}
                          onChange={(e) => setFormData({ ...formData, description: e.target.value })}
                          className="mt-1 min-h-[120px]"
                          placeholder={t("offers.profileDescriptionPlaceholder")}
                        />
                      </div>
                      <div>
                        <Label htmlFor="tags" className="text-sm font-medium">Tags (séparés par des virgules)</Label>
                        <Input
                          id="tags"
                          value={Array.isArray(formData.tags) ? formData.tags.join(', ') : ''}
                          onChange={(e) => setFormData({ 
                            ...formData, 
                            tags: e.target.value.split(',').map(tag => tag.trim()).filter(Boolean)
                          })}
                          className="mt-1"
                          placeholder="design, ui, ux, moderne"
                        />
                      </div>
                    </div>
                  </div>
                ) : (
                  <div className="grid grid-cols-1 lg:grid-cols-2 gap-8">
                    <div className="space-y-6">
                      <div className="flex items-center gap-3 p-3 bg-muted/30 rounded-lg">
                        <FileText className="h-5 w-5 text-muted-foreground" />
                        <div>
                          <p className="text-sm text-muted-foreground">{t("description", { ns: "common" })}</p>
                          <p className="font-medium">{offer.description || t("offers.profileNoDescription")}</p>
                        </div>
                      </div>
                      <div className="flex items-center gap-3 p-3 bg-muted/30 rounded-lg">
                        <CreditCard className="h-5 w-5 text-muted-foreground" />
                        <div>
                          <p className="text-sm text-muted-foreground">{t("price", { ns: "common" })}</p>
                          <p className="font-medium">{offer.price ? `${offer.price} €` : t("offers.profileNotSpecified")}</p>
                        </div>
                      </div>
                    </div>
                    <div className="space-y-6">
                      <div className="flex items-center gap-3 p-3 bg-muted/30 rounded-lg">
                        <Tag className="h-5 w-5 text-muted-foreground" />
                        <div>
                          <p className="text-sm text-muted-foreground">{t("category", { ns: "common" })}</p>
                          <p className="font-medium">{offer.category || t("offers.profileUncategorized")}</p>
                        </div>
                      </div>
                      <div className="flex items-center gap-3 p-3 bg-muted/30 rounded-lg">
                        <Clock className="h-5 w-5 text-muted-foreground" />
                        <div>
                          <p className="text-sm text-muted-foreground">{t("offers.validUntil")}</p>
                          <p className="font-medium">
                            {formatDateForDisplay(offer.validUntil)}
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                )}
              </CardContent>
            </Card>
          </TabsContent>

          <TabsContent value="clients">
            <Card>
              <CardHeader>
                <CardTitle className="flex items-center gap-2">
                  <User className="h-5 w-5" />
                  {t("offers.profileClientsTab")}
                </CardTitle>
              </CardHeader>
              <CardContent>
                {offer.clients && offer.clients.length > 0 ? (
                  <ul className="space-y-3">
                    {offer.clients.map((client) => (
                      <li key={client.id} className="flex items-center justify-between p-3 border rounded-md">
                        <div className="flex items-center gap-3 flex-1">
                          <Avatar className="h-10 w-10">
                            <AvatarImage src={client.avatar || "/placeholder.svg"} />
                            <AvatarFallback>{client.name[0]}</AvatarFallback>
                          </Avatar>
                          <div className="flex-1">
                            <Link to={`/agency/clients/${client.id}`} className="font-medium hover:underline">
                              {client.name}
                            </Link>
                            <div className="text-sm text-muted-foreground">
                              <div>{t("company", { ns: "clients" })}: {client.company}</div>
                              <div>{client.email}</div>
                              <div>{client.phone}</div>
                            </div>
                          </div>
                        </div>
                        <Button
                          variant="destructive"
                          size="sm"
                          onClick={() => setClientToRemove({client, offer})}
                        >
                          {t("delete", { ns: "common" })}
                        </Button>
                      </li>
                    ))}
                  </ul>
                ) : (
                  <p className="text-muted-foreground">{t("offers.profileNoClients")}</p>
                )}
              </CardContent>
            </Card>
          </TabsContent>
        </Tabs>
      </div>
    </AgencyLayout>

    <Dialog open={!!clientToRemove} onOpenChange={() => setClientToRemove(null)}>
      <DialogContent className="sm:max-w-[400px]">
        <DialogHeader>
          <DialogTitle>{t("offers.profileDetachConfirmTitle")}</DialogTitle>
          <DialogDescription>
            {t("offers.profileDetachConfirmDesc", { 
              offer: clientToRemove?.offer.title, 
              client: clientToRemove?.client.name 
            })}
          </DialogDescription>
        </DialogHeader>
        <DialogFooter>
          <Button variant="outline" onClick={() => setClientToRemove(null)}>
            {t("cancel")}
          </Button>
          <Button
            variant="destructive"
            className="hover:bg-destructive/90 active:bg-destructive/70 transition-colors"
            onClick={() => {
              if (clientToRemove) {
                fetch(`http://127.0.0.1:8000/api/offers/${clientToRemove.offer.id}/detach-client/${clientToRemove.client.id}`, {
                  method: "POST",
                  headers: {
                    "Content-Type": "application/json",
                    "Authorization": `Bearer ${localStorage.getItem('token')}`
                  }
                })
                .then(() => {
                  const fetchOffer = async () => {
                    const res = await axios.get<Offer>(`${API_URL}/${id}`);
                    setOffer(res.data);
                  };
                  fetchOffer();
                  toast({
                    title: t("offers.profileSaleCancelled"),
                    description: t("offers.profileSaleCancelledDesc", { 
                      client: clientToRemove.client.name, 
                      offer: clientToRemove.offer.title 
                    })
                  });
                  setClientToRemove(null);
                })
                .catch(err => {
                  console.error("Erreur:", err);
                  toast({
                    title: t("error", { ns: "common" }),
                    description: t("offers.profileDetachError"),
                    variant: "destructive"
                  });
                });
              }
            }}
          >
            {t("offers.profileDetachConfirm")}
          </Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
    </>
  );
}
