// agency-frontend/src/pages/agency/PageProfil.tsx
import { useState, useEffect } from "react";
import { useAuth } from "../../context/AuthContext";
import { Card, CardHeader, CardTitle, CardContent, CardDescription } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Avatar, AvatarImage, AvatarFallback } from "@/components/ui/avatar";
import { useToast } from "@/hooks/use-toast";
import { Textarea } from "@/components/ui/textarea";
import { Eye, EyeOff, User, Mail, Phone, MapPin, Calendar, Globe, Lock, Trash2, LogOut, Shield, Camera, Upload, X } from "lucide-react";
import { Badge } from "@/components/ui/badge";
import { Separator } from "@/components/ui/separator";
import { Link } from "react-router-dom";
import { ArrowLeft } from "lucide-react";
import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle,
} from "@/components/ui/alert-dialog";
import { useTranslation } from "react-i18next";


type Passwords = {
  current_password: string;
  new_password: string;
  new_password_confirmation: string;
};

export default function PageProfil() {
  const { t } = useTranslation("profil");
  const { user, updateUser, refreshUser } = useAuth();
  const { toast } = useToast();

  const [formData, setFormData] = useState({
    name: user?.name || "",
    contact: user?.contact || "",
    adresse: user?.adresse || "",
    ville: user?.ville || "",
    pays: user?.pays || "",
    date_naissance: user?.date_naissance || "",
    genre: user?.genre || "",
    avatar: user?.avatar || "",
    bio: user?.bio || "",
    reseau_sociaux: user?.reseau_sociaux || "",
    email: user?.email || "",
  });

  const [avatarFile, setAvatarFile] = useState<File | null>(null);
  const [avatarPreview, setAvatarPreview] = useState<string>("");
  const [isUploading, setIsUploading] = useState(false);
  const [showAvatarModal, setShowAvatarModal] = useState(false);
  const [loading, setLoading] = useState(false);
  const [openDialog, setOpenDialog] = useState(false);
  const [deleteDialog, setDeleteDialog] = useState(false);
  const [deleteConfirmText, setDeleteConfirmText] = useState("");

  const [passwords, setPasswords] = useState<Passwords>({
    current_password: "",
    new_password: "",
    new_password_confirmation: "",
  });
  const [changingPassword, setChangingPassword] = useState(false);
  const [showPasswordForm, setShowPasswordForm] = useState(false);
  const [showCurrent, setShowCurrent] = useState(false);
  const [showNew, setShowNew] = useState(false);
  const [showConfirm, setShowConfirm] = useState(false);

  // ✅ CORRECTION : Synchronisation améliorée avec l'utilisateur
  useEffect(() => {
    if (user) {
      setFormData({
        name: user.name || "",
        contact: user.contact || "",
        adresse: user.adresse || "",
        ville: user.ville || "",
        pays: user.pays || "",
        date_naissance: user.date_naissance || "",
        genre: user.genre || "",
        avatar: user.avatar || "",
        bio: user.bio || "",
        reseau_sociaux: user.reseau_sociaux || "",
        email: user.email || "",
      });
      setAvatarPreview(user.avatar || "");
    }
  }, [user]);

  const handleAvatarSelect = (event: React.ChangeEvent<HTMLInputElement>) => {
    const file = event.target.files?.[0];
    if (file) {
      if (!file.type.startsWith('image/')) {
        toast({
          title: t("toast.invalidFormat.title"),
          description: t("toast.invalidFormat.description"),
          variant: "destructive",
        });
        
        return;
      }

      if (file.size > 5 * 1024 * 1024) {
        toast({
          title: t("toast.tooLarge.title"),
          description: t("toast.tooLarge.description"),
          variant: "destructive",
        });
        
        return;
      }

      setAvatarFile(file);
      
      const reader = new FileReader();
      reader.onload = (e) => {
        setAvatarPreview(e.target?.result as string);
      };
      reader.readAsDataURL(file);
      
      setShowAvatarModal(true);
    }
  };

  // ✅ CORRECTION : Fonction upload améliorée
  const uploadAvatar = async () => {
    if (!avatarFile) return;

    setIsUploading(true);
    
    try {
      const formData = new FormData();
      formData.append('avatar', avatarFile);
      
      const token = localStorage.getItem("token");
      const response = await fetch('http://127.0.0.1:8000/api/upload-avatar', {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
        },
        body: formData,
      });

      if (!response.ok) {
        const errorData = await response.json();
        throw new Error(errorData.message || 'Erreur lors de l\'upload');
      }

      const data = await response.json();
      
      // ✅ FORCER la mise à jour immédiate
      await refreshUser();
      
      // ✅ Mettre à jour l'état local
      setFormData(prev => ({ ...prev, avatar: data.avatarUrl }));
      setAvatarPreview(data.avatarUrl);
      
      toast({
        title: t("toast.avatarUpdated.title"),
        description: t("toast.avatarUpdated.description"),
        className: "bg-green-50 text-green-800 border-green-200",
      });      
      
      setShowAvatarModal(false);
      setAvatarFile(null);
      
    } catch (error: any) {
      console.error('Upload error:', error);
      toast({
        title: t("toast.avatarError.title"),
        description: error.message || t("toast.avatarError.description"),
        variant: "destructive",
      });
      
    } finally {
      setIsUploading(false);
    }
  };

  // ✅ CORRECTION : Fonction remove améliorée
  const removeAvatar = async () => {
    try {
      const token = localStorage.getItem("token");
      const response = await fetch('http://127.0.0.1:8000/api/remove-avatar', {
        method: 'DELETE',
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': 'application/json',
        },
      });

      if (!response.ok) {
        const errorData = await response.json();
        throw new Error(errorData.message || t("toast.deleteError.description"));
      }

      // ✅ FORCER la mise à jour immédiate
      await refreshUser();
      
      // ✅ Mettre à jour l'état local
      setFormData(prev => ({ ...prev, avatar: "" }));
      setAvatarPreview("");
      
      toast({
        title: t("toast.avatarDeleted.title"),
        description: t("toast.avatarDeleted.description"),
        className: "bg-green-50 text-green-800 border-green-200",
      });
      
      
    } catch (error: any) {
      console.error('Remove error:', error);
      toast({
        title: t("toast.avatarDeleteError.title"),
        description: t("toast.avatarDeleteError.description"),
        variant: "destructive",
      });
      
    }
  };

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    setLoading(true);
    try {
      await updateUser(formData);
      toast({
        title: t("toast.profileUpdated.title"),
        description: t("toast.profileUpdated.description"),
        className: "bg-green-50 text-green-800 border-green-200",
      });
      
    } catch {
      toast({
        title: t("toast.updateError.title"),
        description: t("toast.updateError.description"),
        variant: "destructive",
      });
      
    } finally {
      setLoading(false);
    }
  };

  const handleChangePassword = async (e: React.FormEvent) => {
    e.preventDefault();
    setChangingPassword(true);

    try {
      const token = localStorage.getItem("token");
      if (!token) throw new Error(t("toast.tokenMissing.description"));

      const res = await fetch("http://127.0.0.1:8000/api/change-password", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
        body: JSON.stringify(passwords),
      });

      if (!res.ok) {
        const data = await res.json();
        throw new Error(data.message || t("toast.passwordError.description"));
      }

      toast({
        title: t("toast.passwordUpdated.title"),
        description: t("toast.passwordUpdated.description"),
        className: "bg-green-50 text-green-800 border-green-200",
      });
      

      setPasswords({
        current_password: "",
        new_password: "",
        new_password_confirmation: "",
      });
      setShowPasswordForm(false);
    } catch (err: unknown) {
      toast({
        title: t("toast.passwordError.title"),
        description: t("toast.passwordError.description"),
        variant: "destructive",
      });
      
    } finally {
      setChangingPassword(false);
    }
  };

  const handleDeleteAccount = async () => {
    try {
      const token = localStorage.getItem("token");
      if (!token) throw new Error(t("toast.tokenMissing.description"));
      const res = await fetch("http://127.0.0.1:8000/api/delete-account", {
        method: "DELETE",
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });

      if (!res.ok) {
        const data = await res.json();
        throw new Error(t("toast.accountDeleteError.description"));
      }

      toast({
        title: t("toast.accountDeleted.title"),
        description: t("toast.accountDeleted.description"),
        className: "bg-red-50 text-red-800 border-red-200",
      });
      

      localStorage.removeItem("token");
      localStorage.removeItem("user");
      localStorage.removeItem("isAuthenticated");
      window.location.href = "/";
    } catch (err: unknown) {
      toast({
        title: t("toast.accountDeleteError.title"),
        description: t("toast.accountDeleteError.description"),
        variant: "destructive",
      });      
    }
  };

  const handleDeleteConfirm = () => {
    if (deleteConfirmText === "SUPPRIMER") {
      setDeleteDialog(false);
      setDeleteConfirmText("");
      handleDeleteAccount();
    }
  };

  return (
    <div className="min-h-screen bg-gradient-to-br from-slate-50 to-blue-50/30 py-8 px-4">
      <div className="max-w-6xl mx-auto">
      <div className="flex justify-start mb-6">
        <Button asChild variant="outline" className="shadow-sm">
          <Link to="/agency">
            <ArrowLeft className="mr-2 h-4 w-4" />
            {t("backDashboard")}
          </Link>
        </Button>
      </div>
        <div className="text-center mb-8">
          <h1 className="text-3xl font-bold text-gray-900 mb-2">{t("pageTitle")}</h1>
          <p className="text-gray-600">{t("pageSubtitle")}</p>
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-4 gap-8">
          <div className="lg:col-span-1 space-y-6">
            <Card className="p-6 text-center shadow-lg border-0 bg-white/80 backdrop-blur-sm">
              <div className="relative inline-block group">
                <Avatar className="w-32 h-32 border-4 border-white shadow-xl cursor-pointer transition-all duration-300 group-hover:scale-105">
                  {avatarPreview ? (
                    <AvatarImage 
                      src={avatarPreview} 
                      alt="Profil" 
                      className="object-cover"
                    />
                  ) : (
                    <AvatarFallback className="bg-gradient-to-br from-blue-500 to-purple-600 text-white text-2xl font-semibold">
                      {formData.name?.charAt(0) || "U"}
                    </AvatarFallback>
                  )}
                </Avatar>
                
                <label htmlFor="avatar-upload" className="absolute bottom-2 right-2 bg-blue-500 rounded-full p-2 shadow-lg cursor-pointer transition-all hover:bg-blue-600 hover:scale-110">
                  <Camera className="h-4 w-4 text-white" />
                  <input
                    id="avatar-upload"
                    type="file"
                    accept="image/*"
                    onChange={handleAvatarSelect}
                    className="hidden"
                  />
                </label>
                
                <div className="absolute inset-0 bg-black bg-opacity-0 group-hover:bg-opacity-20 rounded-full transition-all duration-300 flex items-center justify-center">
                  <span className="text-white text-sm font-medium opacity-0 group-hover:opacity-100 transition-opacity">
                  {t("avatarChange")}
                  </span>
                </div>
              </div>
              
              <h2 className="mt-6 text-xl font-bold text-gray-900">
                {formData.name || "Utilisateur"}
              </h2>
              <p className="text-gray-500 text-sm mt-1">{formData.email}</p>
              
              <Badge variant="secondary" className="mt-3 bg-blue-100 text-blue-700">
                <User className="h-3 w-3 mr-1" />
                {t("userRole")}
              </Badge>
              
              {avatarPreview && (
                <Button
                  variant="ghost"
                  size="sm"
                  onClick={removeAvatar}
                  className="mt-3 text-red-600 hover:text-red-700 hover:bg-red-50"
                >
                  <X className="h-3 w-3 mr-1" />
                  {t("avatarRemoved")}
                </Button>
              )}
              
              <div className="mt-6 space-y-3 text-sm text-gray-600">
                <div className="flex items-center justify-center gap-2">
                  <Calendar className="h-4 w-4" />
                  <span>{t("memberSince")}</span>
                </div>
                <div className="flex items-center justify-center gap-2">
                  <Shield className="h-4 w-4 text-green-500" />
                  <span>{t("verifiedAccount")}</span>
                </div>
              </div>
            </Card>

            <Card className="p-6 shadow-lg border-0 bg-white/80 backdrop-blur-sm">
              <h3 className="font-semibold text-gray-900 mb-4">{t("quickActions")}</h3>
              <div className="space-y-3">
                <Button variant="outline" className="w-full justify-start gap-2">
                  <User className="h-4 w-4" />
                  {t("myActivity")}
                </Button>
                <Button 
                  variant="outline" 
                  className="w-full justify-start gap-2 text-red-600 hover:text-red-700 hover:bg-red-50"
                  onClick={() => setOpenDialog(true)}
                >
                  <LogOut className="h-4 w-4" />
                  {t("logoutTitle")}
                </Button>
              </div>
            </Card>
          </div>

          <div className="lg:col-span-3 space-y-6">
            <Card className="shadow-lg border-0 overflow-hidden">
              <CardHeader className="bg-gradient-to-r from-blue-500 to-blue-600 text-white pb-4">
                <CardTitle className="flex items-center gap-2">
                  <User className="h-5 w-5" />
                  {t("personalInfo")}
                </CardTitle>
                <CardDescription className="text-blue-100">
                {t("personalInfoDesc")}
                </CardDescription>
              </CardHeader>
              <CardContent className="p-6">
                <form onSubmit={handleSubmit} className="space-y-6">
                  <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
                    <div className="space-y-2">
                      <label className="text-sm font-medium text-gray-700 flex items-center gap-2">
                        <User className="h-4 w-4" />
                        {t("nameLabel")}
                      </label>
                      <Input
                        value={formData.name}
                        onChange={(e) => setFormData({ ...formData, name: e.target.value })}
                        placeholder={t("nameLabel")}
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>

                    <div className="space-y-2">
                      <label className="text-sm font-medium text-gray-700 flex items-center gap-2">
                        <Mail className="h-4 w-4" />
                        {t("emailLabel")}
                      </label>
                      <Input
                        type="email"
                        value={formData.email}
                        onChange={(e) => setFormData({ ...formData, email: e.target.value })}
                        placeholder="votre@email.com"
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>

                    <div className="space-y-2">
                      <label className="text-sm font-medium text-gray-700 flex items-center gap-2">
                        <Phone className="h-4 w-4" />
                        {t("phoneLabel")}
                      </label>
                      <Input
                        value={formData.contact}
                        onChange={(e) => setFormData({ ...formData, contact: e.target.value })}
                        placeholder="+33 1 23 45 67 89"
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>

                    <div className="space-y-2">
                      <label className="text-sm font-medium text-gray-700 flex items-center gap-2">
                        <Calendar className="h-4 w-4" />
                        {t("birthDateLabel")}
                      </label>
                      <Input
                        type="date"
                        value={formData.date_naissance}
                        onChange={(e) => setFormData({ ...formData, date_naissance: e.target.value })}
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>

                    <div className="md:col-span-2 space-y-2">
                      <label className="text-sm font-medium text-gray-700 flex items-center gap-2">
                        <MapPin className="h-4 w-4" />
                        {t("addressLabel")}
                      </label>
                      <Input
                        value={formData.adresse}
                        onChange={(e) => setFormData({ ...formData, adresse: e.target.value })}
                        // placeholder="Votre adresse complète"
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>

                    <div className="space-y-2">
                      <label className="text-sm font-medium text-gray-700">{t("cityLabel")}</label>
                      <Input
                        value={formData.ville}
                        onChange={(e) => setFormData({ ...formData, ville: e.target.value })}
                        // placeholder={t("cityLabel")}
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>

                    <div className="space-y-2">
                      <label className="text-sm font-medium text-gray-700">{t("countryLabel")}</label>
                      <Input
                        value={formData.pays}
                        onChange={(e) => setFormData({ ...formData, pays: e.target.value })}
                        // placeholder="Votre pays"
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>

                    <div className="md:col-span-2 space-y-2">
                      <label className="text-sm font-medium text-gray-700">{t("avatarUrlLabel")}</label>
                      <div className="flex gap-2">
                        <Input
                          value={formData.avatar}
                          onChange={(e) => setFormData({ ...formData, avatar: e.target.value })}
                          placeholder="https://example.com/photo.jpg"
                          className="focus:ring-2 focus:ring-blue-500"
                        />
                        <Button 
                          type="button" 
                          variant="outline"
                          onClick={() => setAvatarPreview(formData.avatar)}
                        >
                         {t("avatarPreviewButton")}
                        </Button>
                      </div>
                      <p className="text-xs text-gray-500">
                      {t("avatarHint")}
                      </p>
                    </div>

                    <div className="md:col-span-2 space-y-2">
                      <label className="text-sm font-medium text-gray-700">{t("bioLabel")}</label>
                      <Textarea
                        value={formData.bio}
                        onChange={(e) => setFormData({ ...formData, bio: e.target.value })}
                        placeholder={t("bioPlaceholder")}
                        rows={4}
                        className="focus:ring-2 focus:ring-blue-500 resize-none"
                      />
                    </div>

                    <div className="md:col-span-2 space-y-2">
                      <label className="text-sm font-medium text-gray-700 flex items-center gap-2">
                        <Globe className="h-4 w-4" />
                        {t("socialLabel")}
                      </label>
                      <Input
                        value={formData.reseau_sociaux}
                        onChange={(e) => setFormData({ ...formData, reseau_sociaux: e.target.value })}
                        placeholder="https://linkedin.com/in/votre-profil"
                        className="focus:ring-2 focus:ring-blue-500"
                      />
                    </div>
                  </div>

                  <Separator />

                  <div className="flex justify-end">
                    <Button
                      type="submit"
                      disabled={loading}
                      className="bg-blue-600 hover:bg-blue-700 text-white px-8 shadow-lg shadow-blue-500/25"
                    >
                      {loading ? (
                        <>
                          <div className="animate-spin rounded-full h-4 w-4 border-b-2 border-white mr-2"></div>
                          {t("saving")}
                        </>
                      ) : (
                        t("saveButton")
                      )}
                    </Button>
                  </div>
                </form>
              </CardContent>
            </Card>

            <Card className="shadow-lg border-0 overflow-hidden">
              <CardHeader className="bg-gradient-to-r from-green-500 to-green-600 text-white pb-4">
                <CardTitle className="flex items-center gap-2">
                  <Shield className="h-5 w-5" />
                  {t("passwordSection")}
                </CardTitle>
                <CardDescription className="text-green-100">
                {t("securityInfoDesc")}
                </CardDescription>
              </CardHeader>
              <CardContent className="p-6 space-y-6">
                <div className="flex items-center justify-between p-4 bg-gray-50 rounded-lg border">
                  <div className="flex items-center gap-3">
                    <div className="p-2 bg-blue-100 rounded-lg">
                      <Lock className="h-5 w-5 text-blue-600" />
                    </div>
                    <div>
                      <h4 className="font-semibold text-gray-900">{t("passwordTitle")}</h4>
                      <p className="text-sm text-gray-600">{t("passwordDesc")}</p>
                    </div>
                  </div>
                  <Button
                    variant={showPasswordForm ? "secondary" : "outline"}
                    onClick={() => setShowPasswordForm(!showPasswordForm)}
                  >
                    {showPasswordForm ? "Annuler" : "Modifier"}
                  </Button>
                </div>

                {showPasswordForm && (
                  <form onSubmit={handleChangePassword} className="space-y-4 bg-blue-50/50 p-6 rounded-lg border border-blue-200">
                    <h4 className="font-semibold text-gray-900 mb-4">{t("passwordChange")}</h4>
                    
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                      <div className="space-y-2">
                        <label className="text-sm font-medium text-gray-700">{t("passwordCurrent")}</label>
                        <div className="relative">
                          <Input
                            type={showCurrent ? "text" : "password"}
                            value={passwords.current_password}
                            onChange={(e) => setPasswords({ ...passwords, current_password: e.target.value })}
                            className="focus:ring-2 focus:ring-blue-500"
                          />
                          <button
                            type="button"
                            className="absolute inset-y-0 right-3 flex items-center text-gray-500 hover:text-gray-700"
                            onClick={() => setShowCurrent(!showCurrent)}
                          >
                            {showCurrent ? <EyeOff size={18} /> : <Eye size={18} />}
                          </button>
                        </div>
                      </div>

                      <div className="space-y-2">
                        <label className="text-sm font-medium text-gray-700">{t("passwordNew")}</label>
                        <div className="relative">
                          <Input
                            type={showNew ? "text" : "password"}
                            value={passwords.new_password}
                            onChange={(e) => setPasswords({ ...passwords, new_password: e.target.value })}
                            className="focus:ring-2 focus:ring-blue-500"
                          />
                          <button
                            type="button"
                            className="absolute inset-y-0 right-3 flex items-center text-gray-500 hover:text-gray-700"
                            onClick={() => setShowNew(!showNew)}
                          >
                            {showNew ? <EyeOff size={18} /> : <Eye size={18} />}
                          </button>
                        </div>
                      </div>

                      <div className="space-y-2 md:col-span-2">
                        <label className="text-sm font-medium text-gray-700">{t("passwordConfirm")}</label>
                        <div className="relative">
                          <Input
                            type={showConfirm ? "text" : "password"}
                            value={passwords.new_password_confirmation}
                            onChange={(e) => setPasswords({ ...passwords, new_password_confirmation: e.target.value })}
                            className="focus:ring-2 focus:ring-blue-500"
                          />
                          <button
                            type="button"
                            className="absolute inset-y-0 right-3 flex items-center text-gray-500 hover:text-gray-700"
                            onClick={() => setShowConfirm(!showConfirm)}
                          >
                            {showConfirm ? <EyeOff size={18} /> : <Eye size={18} />}
                          </button>
                        </div>
                      </div>
                    </div>

                    <div className="flex gap-3 pt-2">
                      <Button
                        type="submit"
                        disabled={changingPassword}
                        className="bg-green-600 hover:bg-green-700 text-white shadow-lg shadow-green-500/25"
                      >
                        {changingPassword ? (
                          <>
                            <div className="animate-spin rounded-full h-4 w-4 border-b-2 border-white mr-2"></div>
                            {t("passwordSaving")}
                          </>
                        ) : (
                          t("passwordSave")
                        )}
                      </Button>
                    </div>
                  </form>
                )}

                <div className="flex items-center justify-between p-4 bg-red-50 rounded-lg border border-red-200">
                  <div className="flex items-center gap-3">
                    <div className="p-2 bg-red-100 rounded-lg">
                      <Trash2 className="h-5 w-5 text-red-600" />
                    </div>
                    <div>
                      <h4 className="font-semibold text-gray-900">{t("deleteAccountTitle")}</h4>
                      <p className="text-sm text-gray-600">{t("deleteAccountDesc")}</p>
                    </div>
                  </div>
                  <Button
                    variant="destructive"
                    onClick={() => setDeleteDialog(true)}
                    className="bg-red-600 hover:bg-red-700 text-white shadow-lg shadow-red-500/25"
                  >
                    <Trash2 className="h-4 w-4 mr-2" />
                    {t("deleteButton")}
                  </Button>
                </div>
              </CardContent>
            </Card>
          </div>
        </div>
      </div>

      <AlertDialog open={showAvatarModal} onOpenChange={setShowAvatarModal}>
        <AlertDialogContent className="border-0 shadow-xl max-w-md">
          <AlertDialogHeader>
            <AlertDialogTitle className="flex items-center gap-2">
              <Camera className="h-5 w-5 text-blue-600" />
              {t("avatarChange")}
            </AlertDialogTitle>
            <AlertDialogDescription className="space-y-4">
              <div className="flex justify-center">
                <Avatar className="w-24 h-24 border-4 border-blue-200">
                  {avatarPreview ? (
                    <AvatarImage src={avatarPreview} alt="Nouvel avatar" />
                  ) : (
                    <AvatarFallback className="bg-blue-100 text-blue-600">
                      <User className="h-8 w-8" />
                    </AvatarFallback>
                  )}
                </Avatar>
              </div>
              <p className="text-center text-gray-600">
              {t("avatarConfirm")}
              </p>
            </AlertDialogDescription>
          </AlertDialogHeader>
          <AlertDialogFooter className="flex gap-2">
            <AlertDialogCancel 
              onClick={() => {
                setShowAvatarModal(false);
                setAvatarFile(null);
                setAvatarPreview(formData.avatar);
              }}
              className="flex-1"
            >
             {t("cancelButton")}
            </AlertDialogCancel>
            <AlertDialogAction
              onClick={uploadAvatar}
              disabled={isUploading}
              className="flex-1 bg-blue-600 hover:bg-blue-700 text-white"
            >
              {isUploading ? (
                <>
                  <div className="animate-spin rounded-full h-4 w-4 border-b-2 border-white mr-2"></div>
                  Upload...
                </>
              ) : (
                <>
                  <Upload className="h-4 w-4 mr-2" />
                  {t("confirmButton")}
                </>
              )}
            </AlertDialogAction>
          </AlertDialogFooter>
        </AlertDialogContent>
      </AlertDialog>

      <AlertDialog open={openDialog} onOpenChange={setOpenDialog}>
        <AlertDialogContent className="border-0 shadow-xl">
          <AlertDialogHeader>
            <AlertDialogTitle className="flex items-center gap-2 text-gray-900">
              <LogOut className="h-5 w-5 text-blue-600" />
              {t("logoutTitle")}
            </AlertDialogTitle>
            <AlertDialogDescription className="text-gray-600">
            {t("logoutDesc")}
            </AlertDialogDescription>
          </AlertDialogHeader>
          <AlertDialogFooter>
            <AlertDialogCancel className="border-gray-300 hover:bg-gray-50">
            {t("cancelButton")}
            </AlertDialogCancel>
            <AlertDialogAction
              onClick={() => {
                setOpenDialog(false);
                localStorage.removeItem("user");
                localStorage.removeItem("isAuthenticated");
                window.location.href = "/login";
              }}
              className="bg-blue-600 hover:bg-blue-700 text-white"
            >
              {t("logoutButton")}
            </AlertDialogAction>
          </AlertDialogFooter>
        </AlertDialogContent>
      </AlertDialog>

      <AlertDialog open={deleteDialog} onOpenChange={setDeleteDialog}>
        <AlertDialogContent className="border-0 shadow-xl max-w-md">
          <AlertDialogHeader>
            <AlertDialogTitle className="flex items-center gap-2 text-red-700">
              <Trash2 className="h-5 w-5" />
              {t("deleteDialogTitle")}
            </AlertDialogTitle>
            <AlertDialogDescription className="space-y-3 text-gray-600">
              <p>{t("deleteDialogDesc")}
              </p>
              <ul className="list-disc list-inside text-sm space-y-1 text-gray-500">
                <li>{t("deleteDialogList1")}</li>
                <li>{t("deleteDialogList2")}</li>
                <li>{t("deleteDialogList3")}</li>
              </ul>
              <div className="pt-2">
                <label className="text-sm font-medium text-gray-700">
                {t("deleteConfirmText")}
                </label>
                <Input
                  type="text"
                  value={deleteConfirmText}
                  onChange={(e) => setDeleteConfirmText(e.target.value)}
                  placeholder="SUPPRIMER"
                  className="mt-2 border-red-300 focus:border-red-500 focus:ring-red-500"
                />
              </div>
            </AlertDialogDescription>
          </AlertDialogHeader>
          <AlertDialogFooter>
            <AlertDialogCancel 
              onClick={() => {
                setDeleteDialog(false);
                setDeleteConfirmText("");
              }}
              className="border-gray-300 hover:bg-gray-50"
            >
              {t("cancelButton")}
            </AlertDialogCancel>
            <AlertDialogAction
              disabled={deleteConfirmText !== "SUPPRIMER"}
              onClick={handleDeleteConfirm}
              className="bg-red-600 hover:bg-red-700 disabled:opacity-50 disabled:cursor-not-allowed text-white"
            >
              {t("deleteDialogTitle")}
            </AlertDialogAction>
          </AlertDialogFooter>
        </AlertDialogContent>
      </AlertDialog>
    </div>
  );
}
