// agency-frontend/src/pages/agency/Statistic.tsx
import React, { useEffect, useState, useMemo } from 'react';
import { Bar } from 'react-chartjs-2';
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend,
} from 'chart.js';
import { fetchChiffreAffaire, fetchVentesTotales } from '@/api/statistics';
import { StatisticsData } from '@/api/statistics';
import { useTranslation } from 'react-i18next';
import { useAuth } from "@/context/AuthContext";
import { useNavigate } from "react-router-dom";
import { AlertTriangle, ArrowLeft } from "lucide-react";
import { Button } from "@/components/ui/button";

// Enregistrer les composants Chart.js
ChartJS.register(
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend
);

// Composants d'icônes
const BarChart = () => <span>📊</span>;
const TrendingUp = () => <span>📈</span>;

type Period = '12m' | '6m' | 'current_year';

const Statistic: React.FC = () => {
  const { t } = useTranslation("statistics");
  const { hasPermission, isAdmin } = useAuth();
  const navigate = useNavigate();

  // ✅ Vérification des permissions
  const canViewStatistics = isAdmin() || hasPermission('view_statistics');

  const [activeTab, setActiveTab] = useState<'ca' | 'ventes'>('ca');
  const [period, setPeriod] = useState<Period>('12m');
  const [caData, setCaData] = useState<StatisticsData | null>(null);
  const [ventesData, setVentesData] = useState<StatisticsData | null>(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);

  // ✅ SI L'UTILISATEUR N'A PAS LA PERMISSION
  if (!canViewStatistics) {
    return (
      <div className="flex items-center justify-center min-h-[60vh]">
        <div className="text-center max-w-md">
          <AlertTriangle className="h-16 w-16 text-yellow-500 mx-auto mb-4" />
          <h2 className="text-2xl font-bold text-foreground mb-2">Accès restreint</h2>
          <p className="text-muted-foreground mb-6">
            Vous n'avez pas la permission d'accéder aux statistiques.
          </p>
          <div className="flex gap-3 justify-center">
            <Button variant="outline" onClick={() => navigate(-1)}>
              <ArrowLeft className="w-4 h-4 mr-2" />
              Retour
            </Button>
            <Button onClick={() => navigate("/agency")}>
              Accueil
            </Button>
          </div>
        </div>
      </div>
    );
  }

  useEffect(() => {
    const loadStatistics = async () => {
      try {
        setLoading(true);
        const [ca, ventes] = await Promise.all([
          fetchChiffreAffaire(),
          fetchVentesTotales()
        ]);
        setCaData(ca);
        setVentesData(ventes);
      } catch (err) {
        setError(t("error"));
        console.error('Erreur:', err);
      } finally {
        setLoading(false);
      }
    };

    loadStatistics();
  }, []);

  // Calculs des métriques pour CA
  const caMetrics = useMemo(() => {
    if (!caData?.data.length) return null;

    const data = caData.data;
    const totalCA = data.reduce((sum, value) => sum + value, 0);
    const avgMonthlyCA = totalCA / data.length;
    const maxMonthlyCA = Math.max(...data);
    
    const trend = data.length >= 2 
      ? ((data[data.length - 1] - data[data.length - 2]) / data[data.length - 2]) * 100
      : 0;

    return {
      totalCA,
      avgMonthlyCA,
      maxMonthlyCA,
      trend,
      monthCount: data.length
    };
  }, [caData]);

  // Calculs des métriques pour Ventes
  const ventesMetrics = useMemo(() => {
    if (!ventesData?.data.length) return null;

    const data = ventesData.data;
    const totalVentes = data.reduce((sum, value) => sum + value, 0);
    const avgMonthlyVentes = totalVentes / data.length;
    const maxMonthlyVentes = Math.max(...data);
    
    const trend = data.length >= 2 
      ? ((data[data.length - 1] - data[data.length - 2]) / data[data.length - 2]) * 100
      : 0;

    return {
      totalVentes,
      avgMonthlyVentes,
      maxMonthlyVentes,
      trend,
      monthCount: data.length
    };
  }, [ventesData]);

  // Données pour l'histogramme CA
  const caChartData = {
    labels: caData?.labels || [],
    datasets: [
      {
        label: t("charts.caTitle"),
        data: caData?.data || [],
        backgroundColor: 'rgba(59, 130, 246, 0.8)',
        borderColor: 'rgba(59, 130, 246, 1)',
        borderWidth: 1,
      },
    ],
  };

  // Données pour l'histogramme Ventes
  const ventesChartData = {
    labels: ventesData?.labels || [],
    datasets: [
      {
        label: t("charts.ventesYAxisLabel"),
        data: ventesData?.data || [],
        backgroundColor: 'rgba(34, 197, 94, 0.8)',
        borderColor: 'rgba(34, 197, 94, 1)',
        borderWidth: 1,
      },
    ],
  };

  const caChartOptions = {
    responsive: true,
    plugins: {
      legend: {
        position: 'top' as const,
      },
      title: {
        display: true,
        text: t("charts.caTitle"),
      },
      tooltip: {
        callbacks: {
          label: function(context: any) {
            return t("charts.caTooltip", { value: context.parsed.y });
          }
        }
      },
    },
    scales: {
      y: {
        beginAtZero: true,
        title: {
          display: true,
          text: t("charts.caTooltipUnit"),
        },
        ticks: {
          callback: function(value: any) {
            return value.toLocaleString('fr-FR') + ' €';
          }
        }
      },
      x: {
        title: {
          display: true,
          text: t("charts.monthLabel"),
        },
      },
    },
  };

  const ventesChartOptions = {
    responsive: true,
    plugins: {
      legend: {
        position: 'top' as const,
      },
      title: {
        display: true,
        text: t("charts.ventesTitle"),
      },
      tooltip: {
        callbacks: {
          label: function(context: any) {
            return `${t("charts.ventesTooltip")}: ${context.parsed.y}`;
          }
        }
      },
    },
    scales: {
      y: {
        beginAtZero: true,
        title: {
          display: true,
          text: t("charts.ventesYAxisLabel"),
        },
        ticks: {
          callback: function(value: any) {
            return value + ` ${t("charts.salesUnit")}`;
          }
        }
      },
      x: {
        title: {
          display: true,
          text: t("charts.monthLabel"),
        },
      },
    },
  };

  if (loading) {
    return (
      <div className="flex justify-center items-center h-64">
        <div className="text-lg">{t("loading")}</div>
      </div>
    );
  }

  if (error) {
    return (
      <div className="flex justify-center items-center h-64">
        <div className="text-red-500 text-lg">{error}</div>
      </div>
    );
  }

  return (
    <div className="p-6">
      <h1 className="text-2xl font-bold mb-6">{t("title")}</h1>

      {/* Cartes de résumé dynamiques selon l'onglet actif */}
      {activeTab === 'ca' && caMetrics && (
        <div className="grid grid-cols-1 md:grid-cols-3 gap-4 mb-6">
          <div className="bg-white p-4 rounded-lg shadow border-l-4 border-blue-500">
            <div className="text-sm text-gray-500">{t("caMetrics.avgMonthlyCA")}</div>
            <div className="text-2xl font-bold text-blue-600">
              {caMetrics.avgMonthlyCA.toLocaleString('fr-FR', { maximumFractionDigits: 0 })} €
            </div>
            <div className="text-xs text-gray-400 mt-1">{t("caMetrics.monthsAnalyzed", { monthCount: caMetrics.monthCount })}</div>
          </div>

          <div className="bg-white p-4 rounded-lg shadow border-l-4 border-green-500">
            <div className="text-sm text-gray-500">{t("caMetrics.maxMonthlyCA")}</div>
            <div className="text-2xl font-bold text-green-600">
              {caMetrics.maxMonthlyCA.toLocaleString('fr-FR')} €
            </div>
            <div className="text-xs text-gray-400 mt-1">{t("caMetrics.bestPerformance")}</div>
          </div>

          <div className="bg-white p-4 rounded-lg shadow border-l-4 border-purple-500">
            <div className="text-sm text-gray-500">{t("caMetrics.trend")}</div>
            <div className={`text-2xl font-bold ${caMetrics.trend >= 0 ? 'text-green-600' : 'text-red-600'}`}>
              {caMetrics.trend >= 0 ? '↗' : '↘'} {Math.abs(caMetrics.trend).toFixed(1)}%
            </div>
            <div className="text-xs text-gray-400 mt-1">{t("caMetrics.vsLastMonth")}</div>
          </div>
        </div>
      )}

      {activeTab === 'ventes' && ventesMetrics && (
        <div className="grid grid-cols-1 md:grid-cols-3 gap-4 mb-6">
          <div className="bg-white p-4 rounded-lg shadow border-l-4 border-green-500">
            <div className="text-sm text-gray-500">{t("ventesMetrics.avgMonthlyVentes")}</div>
            <div className="text-2xl font-bold text-green-600">
              {ventesMetrics.avgMonthlyVentes.toFixed(0)} {t("charts.ventesPerMonth")}
            </div>
            <div className="text-xs text-gray-400 mt-1">{t("ventesMetrics.monthsAnalyzed", { monthCount: ventesMetrics.monthCount })}</div>
          </div>

          <div className="bg-white p-4 rounded-lg shadow border-l-4 border-blue-500">
            <div className="text-sm text-gray-500">{t("ventesMetrics.maxMonthlyVentes")}</div>
            <div className="text-2xl font-bold text-blue-600">
              {ventesMetrics.maxMonthlyVentes} {t("charts.salesUnit")}
            </div>
            <div className="text-xs text-gray-400 mt-1">{t("ventesMetrics.bestPerformance")}</div>
          </div>

          <div className="bg-white p-4 rounded-lg shadow border-l-4 border-purple-500">
            <div className="text-sm text-gray-500">{t("ventesMetrics.trend")}</div>
            <div className={`text-2xl font-bold ${ventesMetrics.trend >= 0 ? 'text-green-600' : 'text-red-600'}`}>
              {ventesMetrics.trend >= 0 ? '↗' : '↘'} {Math.abs(ventesMetrics.trend).toFixed(1)}%
            </div>
            <div className="text-xs text-gray-400 mt-1">{t("ventesMetrics.vsLastMonth")}</div>
          </div>
        </div>
      )}

      {/* En-tête avec Tabs et Filtres */}
      <div className="flex flex-col lg:flex-row lg:items-center lg:justify-between mb-6 gap-4">
        <div className="flex-1">
          <div className="border-b border-gray-200">
            <nav className="-mb-px flex space-x-8">
              <button
                onClick={() => setActiveTab('ca')}
                className={`py-2 px-1 border-b-2 font-medium text-sm flex items-center gap-2 ${
                  activeTab === 'ca'
                    ? 'border-blue-500 text-blue-600'
                    : 'border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300'
                }`}
              >
                <TrendingUp />
                {t("tabs.ca")}
              </button>
              <button
                onClick={() => setActiveTab('ventes')}
                className={`py-2 px-1 border-b-2 font-medium text-sm flex items-center gap-2 ${
                  activeTab === 'ventes'
                    ? 'border-green-500 text-green-600'
                    : 'border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300'
                }`}
              >
                <BarChart />
                {t("tabs.ventes")}
              </button>
            </nav>
          </div>
        </div>

        <div className="flex items-center gap-3">
          <label htmlFor="period" className="text-sm text-gray-600 whitespace-nowrap">
          {t("period.label")}
          </label>
          <select
            id="period"
            value={period}
            onChange={(e) => setPeriod(e.target.value as Period)}
            className="border border-gray-300 rounded-lg px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"
          >
            <option value="12m">{t("period.12m")}</option>
            <option value="6m">{t("period.6m")}</option>
            <option value="current_year">{t("period.current_year")}</option>
          </select>
        </div>
      </div>

      {/* Contenu des tabs */}
      <div className="bg-white p-6 rounded-lg shadow-md">
        {activeTab === 'ca' ? (
          caData && caData.data.length > 0 ? (
            <>
              <Bar data={caChartData} options={caChartOptions} />
              
              <div className="mt-8 p-4 bg-gradient-to-r from-blue-50 to-indigo-50 rounded-lg border border-blue-200">
                <div className="flex justify-between items-center">
                  <div>
                    <h3 className="text-lg font-semibold text-gray-800">{t("caMetrics.totalCA", { monthCount: caMetrics.monthCount })}</h3>
                    <p className="text-sm text-gray-600">{t("caMetrics.cumulativeCA")}</p>
                  </div>
                  <div className="text-right">
                    <div className="flex items-center justify-end gap-2">
                      <span className="text-2xl font-bold text-blue-700">
                        {caMetrics.totalCA.toLocaleString('fr-FR')} €
                      </span>
                      {caMetrics.trend !== 0 && (
                        <span className={`px-2 py-1 rounded-full text-sm font-medium ${
                          caMetrics.trend >= 0 
                            ? 'bg-green-100 text-green-800' 
                            : 'bg-red-100 text-red-800'
                        }`}>
                          {caMetrics.trend >= 0 ? '↗' : '↘'} {Math.abs(caMetrics.trend).toFixed(1)}%
                        </span>
                      )}
                    </div>
                    <div className="text-sm text-gray-500 mt-1">
                      {t("caMetrics.monthsAnalyzed", { monthCount: caData.data.length })}
                    </div>
                  </div>
                </div>
              </div>
            </>
          ) : (
            <div className="text-center text-gray-500 py-12">
              <div className="text-lg mb-2">{t("noData.ca.title")}</div>
              <div className="text-sm">{t("noData.ca.subtitle")}</div>
            </div>
          )
        ) : (
          ventesData && ventesData.data.length > 0 ? (
            <>
              <Bar data={ventesChartData} options={ventesChartOptions} />
              
              <div className="mt-8 p-4 bg-gradient-to-r from-green-50 to-emerald-50 rounded-lg border border-green-200">
                <div className="flex justify-between items-center">
                  <div>
                    <h3 className="text-lg font-semibold text-gray-800">{t("ventesMetrics.totalVentes", { monthCount: ventesMetrics.monthCount })}</h3>
                    <p className="text-sm text-gray-600">{t("ventesMetrics.totalSales")}</p>
                  </div>
                  <div className="text-right">
                    <div className="flex items-center justify-end gap-2">
                      <span className="text-2xl font-bold text-green-700">
                        {ventesMetrics.totalVentes.toLocaleString('fr-FR')} ventes
                      </span>
                      {ventesMetrics.trend !== 0 && (
                        <span className={`px-2 py-1 rounded-full text-sm font-medium ${
                          ventesMetrics.trend >= 0 
                            ? 'bg-green-100 text-green-800' 
                            : 'bg-red-100 text-red-800'
                        }`}>
                          {ventesMetrics.trend >= 0 ? '↗' : '↘'} {Math.abs(ventesMetrics.trend).toFixed(1)}%
                        </span>
                      )}
                    </div>
                    <div className="text-sm text-gray-500 mt-1">
                    {t("ventesMetrics.monthsAnalyzed", { monthCount: ventesData.data.length })}
                    </div>
                  </div>
                </div>
              </div>
            </>
          ) : (
            <div className="text-center text-gray-500 py-12">
              <div className="text-lg mb-2">{t("noData.ventes.title")}</div>
              <div className="text-sm">{t("noData.ventes.subtitle")}</div>

            </div>
          )
        )}
      </div>
    </div>
  );
};

export default Statistic;
