// agency-frontend/src/components/ui/multiple-image-upload.tsx
import { useState, useRef } from 'react';
import { Button } from '@/components/ui/button';
import { Input } from '@/components/ui/input';
import { Upload, X, Image as ImageIcon, Plus } from 'lucide-react';
import { useToast } from '@/hooks/use-toast';

interface MultipleImageUploadProps {
  value?: string[];
  onChange: (urls: string[]) => void;
  disabled?: boolean;
  maxImages?: number;
}

export function MultipleImageUpload({ 
  value = [], 
  onChange, 
  disabled, 
  maxImages = 5 
}: MultipleImageUploadProps) {
  const [isUploading, setIsUploading] = useState(false);
  const fileInputRef = useRef<HTMLInputElement>(null);
  const { toast } = useToast();

  const handleFileSelect = async (event: React.ChangeEvent<HTMLInputElement>) => {
    const files = Array.from(event.target.files || []);
    if (!files.length) return;

    // Vérifier le nombre maximum d'images
    if (value.length + files.length > maxImages) {
      toast({
        title: "Erreur",
        description: `Vous ne pouvez pas ajouter plus de ${maxImages} images.`,
        variant: "destructive"
      });
      return;
    }

    setIsUploading(true);
    const newUrls: string[] = [];

    try {
      for (const file of files) {
        // Validation du type de fichier
        if (!file.type.startsWith('image/')) {
          toast({
            title: "Erreur",
            description: `${file.name} n'est pas un fichier image valide.`,
            variant: "destructive"
          });
          continue;
        }

        // Validation de la taille (max 5MB)
        if (file.size > 5 * 1024 * 1024) {
          toast({
            title: "Erreur",
            description: `${file.name} ne doit pas dépasser 5MB.`,
            variant: "destructive"
          });
          continue;
        }

        // Créer un FormData pour l'upload
        const formData = new FormData();
        formData.append('image', file);

        // Upload vers le backend
        const response = await fetch('http://127.0.0.1:8000/api/upload-image', {
          method: 'POST',
          body: formData,
          headers: {
            'Authorization': `Bearer ${localStorage.getItem('token')}`
          }
        });

        if (!response.ok) {
          throw new Error(`Erreur lors de l'upload de ${file.name}`);
        }

        const data = await response.json();
        newUrls.push(data.url);
      }

      // Ajouter les nouvelles URLs à la liste existante
      onChange([...value, ...newUrls]);
      
      if (newUrls.length > 0) {
        toast({
          title: "Succès",
          description: `${newUrls.length} image(s) uploadée(s) avec succès.`
        });
      }
    } catch (error) {
      console.error('Erreur upload:', error);
      toast({
        title: "Erreur",
        description: "Erreur lors de l'upload des images.",
        variant: "destructive"
      });
    } finally {
      setIsUploading(false);
      // Reset input
      if (fileInputRef.current) {
        fileInputRef.current.value = '';
      }
    }
  };

  const handleRemove = (indexToRemove: number) => {
    const newUrls = value.filter((_, index) => index !== indexToRemove);
    onChange(newUrls);
  };

  const moveImage = (fromIndex: number, toIndex: number) => {
    const newUrls = [...value];
    const [movedItem] = newUrls.splice(fromIndex, 1);
    newUrls.splice(toIndex, 0, movedItem);
    onChange(newUrls);
  };

  return (
    <div className="space-y-4">
      <div className="flex items-center gap-4">
        <Button
          type="button"
          variant="outline"
          onClick={() => fileInputRef.current?.click()}
          disabled={disabled || isUploading || value.length >= maxImages}
          className="flex items-center gap-2"
        >
          <Upload className="h-4 w-4" />
          {isUploading ? 'Upload en cours...' : 'Ajouter des images'}
        </Button>
        
        <Input
          ref={fileInputRef}
          type="file"
          accept="image/*"
          multiple
          onChange={handleFileSelect}
          className="hidden"
          disabled={disabled || isUploading}
        />
        
        <span className="text-sm text-gray-500">
          {value.length}/{maxImages} images
        </span>
      </div>

      {value.length > 0 && (
        <div className="grid grid-cols-2 md:grid-cols-3 gap-4">
          {value.map((url, index) => (
            <div key={index} className="relative group">
              <div className="relative w-full h-32 rounded-lg border overflow-hidden">
                <img
                  src={url}
                  alt={`Image ${index + 1}`}
                  className="w-full h-full object-cover"
                  onError={(e) => {
                    const target = e.target as HTMLImageElement;
                    target.style.display = 'none';
                  }}
                />
                
                {/* Badge "Première image" */}
                {index === 0 && (
                  <div className="absolute top-2 left-2 bg-green-500 text-white text-xs px-2 py-1 rounded">
                    Image principale
                  </div>
                )}
                
                {/* Bouton supprimer */}
                <Button
                  type="button"
                  variant="destructive"
                  size="sm"
                  onClick={() => handleRemove(index)}
                  disabled={disabled}
                  className="absolute top-2 right-2 h-6 w-6 p-0 opacity-0 group-hover:opacity-100 transition-opacity"
                >
                  <X className="h-3 w-3" />
                </Button>
                
                {/* Boutons de réorganisation */}
                <div className="absolute bottom-2 right-2 flex gap-1 opacity-0 group-hover:opacity-100 transition-opacity">
                  {index > 0 && (
                    <Button
                      type="button"
                      variant="secondary"
                      size="sm"
                      onClick={() => moveImage(index, index - 1)}
                      disabled={disabled}
                      className="h-6 w-6 p-0 text-xs"
                    >
                      ←
                    </Button>
                  )}
                  {index < value.length - 1 && (
                    <Button
                      type="button"
                      variant="secondary"
                      size="sm"
                      onClick={() => moveImage(index, index + 1)}
                      disabled={disabled}
                      className="h-6 w-6 p-0 text-xs"
                    >
                      →
                    </Button>
                  )}
                </div>
              </div>
              
              <p className="text-xs text-center mt-1 text-gray-500">
                {index === 0 ? 'Image principale' : `Image ${index + 1}`}
              </p>
            </div>
          ))}
        </div>
      )}
      
      {value.length === 0 && (
        <div className="border-2 border-dashed border-gray-300 rounded-lg p-8 text-center">
          <ImageIcon className="h-12 w-12 text-gray-400 mx-auto mb-4" />
          <p className="text-gray-500">Aucune image ajoutée</p>
          <p className="text-sm text-gray-400">La première image sera l'image principale</p>
        </div>
      )}
    </div>
  );
}
