// agency-frontend/src/context/AuthContext.tsx
import { createContext, useContext, useState, ReactNode, useEffect } from "react";
import { apiUrl } from "@/lib/config";

export type User = {
  id: number;
  name?: string;
  email: string;
  name_profil?: string;
  lastname_profil?: string;
  contact?: string;
  adresse?: string;
  ville?: string;
  pays?: string;
  date_naissance?: string;
  genre?: string;
  avatar?: string;
  bio?: string;
  reseau_sociaux?: string;
  role?: string;
  permissions?: string[];
};

export type UpdateUserData = Partial<Omit<User, "id">>;

type AuthContextType = {
  user: User | null;
  isLoading: boolean; 
  setAuthUser: (user: User | null) => void;
  login: (email: string, password: string) => Promise<void>;
  logout: () => void;
  updateUser: (data: UpdateUserData) => Promise<User>;
  refreshUser: () => Promise<User>;
  isAdmin: () => boolean;
  hasPermission: (permission: string) => boolean;
  getIsLoading: () => boolean; 
};

const AuthContext = createContext<AuthContextType | undefined>(undefined);

export const AuthProvider = ({ children }: { children: ReactNode }) => {
  const [user, setUser] = useState<User | null>(
    JSON.parse(localStorage.getItem("user") || "null")
  );
  const [isLoading, setIsLoading] = useState(true);

  const getIsLoading = () => isLoading;


  useEffect(() => {
    const token = localStorage.getItem("token");
    if (token && !user) {
      // Utiliser le même endpoint que refreshUser
      fetch(apiUrl("/user"), {
        headers: { 
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}` 
        },
      })
        .then((res) => (res.ok ? res.json() : null))
        .then((data: User | null) => {
          if (data) {
            setUser(data);
            localStorage.setItem("user", JSON.stringify(data));
          }
        })
        .catch(() => {
          setUser(null);
          localStorage.removeItem("user");
        });
    }
  }, [user]);
 
  const setAuthUser = (userData: User | null) => {
    setUser(userData);
    if (userData) {
      localStorage.setItem("user", JSON.stringify(userData));
    } else {
      localStorage.removeItem("user");
    }
  };

  const login = async (email: string, password: string) => {
    const res = await fetch(apiUrl("/login"), {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ email, password }),
    });

    if (!res.ok) {
      throw new Error("Échec de la connexion");
    }

    const data: { user: User; token: string } = await res.json();
    setUser(data.user);
    localStorage.setItem("token", data.token);
    localStorage.setItem("user", JSON.stringify(data.user));
  };

  const logout = () => {
    setUser(null);
    localStorage.removeItem("token");
    localStorage.removeItem("user");
  };

  const updateUser = async (data: UpdateUserData): Promise<User> => {
    const token = localStorage.getItem("token");
    if (!token) {
      throw new Error("Authentification requise (token manquant).");
    }

    const res = await fetch(apiUrl("/user/profile"), {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(data),
    });

    if (!res.ok) {
      const errorData: { message?: string; error?: string } = await res.json();
      throw new Error(
        errorData.message ?? errorData.error ?? "Erreur lors de la mise à jour du profil"
      );
    }

    const updated: User = await res.json();
    setUser(updated);
    localStorage.setItem("user", JSON.stringify(updated));
    return updated;
  };

  const refreshUser = async (): Promise<User> => {
    const token = localStorage.getItem("token");
    if (!token) {
      throw new Error("Token manquant");
    }

    try {
      const res = await fetch(apiUrl("/user"), {
        headers: { 
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}` 
        },
      });

      if (!res.ok) {
        throw new Error(`Erreur HTTP: ${res.status}`);
      }

      const userData: User = await res.json();
      setUser(userData);
      localStorage.setItem("user", JSON.stringify(userData));
      return userData;
    } catch (error) {
      console.error("Erreur refreshUser:", error);
      // En cas d'erreur, on retourne l'utilisateur actuel plutôt que de tout casser
      if (user) {
        return user;
      }
      throw error;
    }
  };

  const isAdmin = (): boolean => {
    return user?.role === 'admin';
  };

  const hasPermission = (permission: string): boolean => {
    if (!user) return false;
    if (isAdmin()) return true;
    return user.permissions?.includes(permission) || false;
  };

  return (
    <AuthContext.Provider value={{ 
      user, 
      isLoading,
      setAuthUser, 
      login, 
      logout, 
      updateUser, 
      refreshUser,
      isAdmin,
      hasPermission,
      getIsLoading
    }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  const context = useContext(AuthContext);
  if (!context) {
    throw new Error("useAuth doit être utilisé dans AuthProvider");
  }
  return context;
};
