// agency-frontend/src/context/SettingsContext.tsx
// src/context/SettingsContext.tsx
import { createContext, useContext, useState, useEffect, ReactNode } from "react";
import { useTranslation } from "react-i18next";
import dayjs from "dayjs";

type SettingsContextType = {
  language: string;
  timezone: string;
  setLanguage: (lang: string) => void;
  setTimezone: (tz: string) => void;
};

const SettingsContext = createContext<SettingsContextType | null>(null);

export const SettingsProvider = ({ children }: { children: ReactNode }) => {
  const { i18n } = useTranslation();
  const [language, setLanguageState] = useState(() => localStorage.getItem("lang") || "fr");
  const [timezone, setTimezoneState] = useState(() => localStorage.getItem("tz") || "Europe/Paris");

  useEffect(() => {
    i18n.changeLanguage(language);
    localStorage.setItem("lang", language);
  }, [language, i18n]);

  useEffect(() => {
    localStorage.setItem("tz", timezone);
    dayjs.tz.setDefault(timezone);
  }, [timezone]);

  const setLanguage = (lang: string) => setLanguageState(lang);
  const setTimezone = (tz: string) => setTimezoneState(tz);

  return (
    <SettingsContext.Provider value={{ language, timezone, setLanguage, setTimezone }}>
      {children}
    </SettingsContext.Provider>
  );
};

export const useSettings = () => {
  const context = useContext(SettingsContext);
  if (!context) throw new Error("useSettings must be used within SettingsProvider");
  return context;
};
