// agency-frontend/src/lib/agency-api.ts
import { Agency, AgencyFormData } from "@/types/agency";
import { apiUrl } from "@/lib/config";

const API_URL = apiUrl("/agency");

export async function fetchAgency(): Promise<Agency> {
  const token = localStorage.getItem("token");
  const res = await fetch(API_URL, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  if (!res.ok) {
    throw new Error("Erreur lors de la récupération de l'agence");
  }

  return res.json();
}

export async function createOrUpdateAgency(data: AgencyFormData): Promise<{ message: string; agency: Agency }> {
  const token = localStorage.getItem("token");
  const res = await fetch(API_URL, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify(data),
  });

  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.message || "Erreur lors de la sauvegarde");
  }

  return res.json();
}

export async function updateAgency(data: Partial<AgencyFormData>): Promise<{ message: string; agency: Agency }> {
  const token = localStorage.getItem("token");
  const res = await fetch(API_URL, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify(data),
  });

  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.message || "Erreur lors de la mise à jour");
  }

  return res.json();
}
