// agency-frontend/src/lib/invitation-api.ts
import { Collaborator, InvitationData, Permission } from '@/types/invitation';

const API_URL = 'http://127.0.0.1:8000/api';

const getAuthHeaders = () => {
  const token = localStorage.getItem('token');
  return {
    'Authorization': `Bearer ${token}`,
    'Content-Type': 'application/json',
  };
};

export const fetchCollaborators = async (): Promise<Collaborator[]> => {
  const response = await fetch(`${API_URL}/collaborators`, {
    headers: getAuthHeaders(),
  });

  if (!response.ok) {
    throw new Error('Erreur lors du chargement des collaborateurs');
  }

  return response.json();
};

export const inviteCollaborator = async (data: InvitationData): Promise<void> => {
  const response = await fetch(`${API_URL}/invite-collaborator`, {
    method: 'POST',
    headers: getAuthHeaders(),
    body: JSON.stringify(data),
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.message || 'Erreur lors de l\'invitation');
  }
};

export const updateCollaboratorPermissions = async (
  userId: number, 
  permissions: Permission[]
): Promise<void> => {
  const response = await fetch(`${API_URL}/collaborators/${userId}/permissions`, {
    method: 'PUT',
    headers: getAuthHeaders(),
    body: JSON.stringify({ permissions }),
  });

  if (!response.ok) {
    throw new Error('Erreur lors de la mise à jour des permissions');
  }
};

export const deleteCollaborator = async (userId: number): Promise<void> => {
  const response = await fetch(`${API_URL}/collaborators/${userId}`, {
    method: 'DELETE',
    headers: getAuthHeaders(),
  });

  if (!response.ok) {
    throw new Error('Erreur lors de la suppression du collaborateur');
  }
};
