// agency-frontend/src/components/agency/AgencyHeader.tsx
import { useState, useEffect } from "react";
import { Search, Bell, User, Settings, LogOut, Mail, Calendar, CheckCircle, AlertTriangle, Info, FileText, Users, CreditCard, Clock, Eye } from "lucide-react";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Badge } from "@/components/ui/badge";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";
import { Avatar, AvatarImage, AvatarFallback } from "@/components/ui/avatar";
import { SidebarTrigger } from "@/components/ui/sidebar";
import { Link, useNavigate } from "react-router-dom";
import { useToast } from "@/hooks/use-toast";
import axios from "axios";

// ✅ Import pour la popup de confirmation
import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle,
} from "@/components/ui/alert-dialog";
import { useTranslation } from "react-i18next";

// Types de notifications basés sur votre modèle Laravel
interface Notification {
  id: number;
  user_id: number;
  type: 'new_client' | 'offer_expiring' | 'payment_received' | 'meeting_reminder' | 'task_assigned' | 'system_alert';
  title: string;
  message: string;
  data: any;
  action_url: string | null;
  read_at: string | null;
  created_at: string;
  updated_at: string;
}

// Types pour les réponses API
interface NotificationsResponse {
  data: Notification[];
}

interface UnreadCountResponse {
  count: number;
}

interface ApiResponse<T> {
  data: T;
}

// Types pour la recherche
interface SearchClient {
  id: number;
  name: string;
  company: string;
  email: string;
}

interface SearchOffer {
  id: number;
  title: string;
  description: string;
  category: string;
  client?: string;
}

export default function AgencyHeader() {
  const { t } = useTranslation("AgencyHeader");
  const [searchQuery, setSearchQuery] = useState("");
  const [searchResults, setSearchResults] = useState<{
    clients: SearchClient[];
    offres: SearchOffer[];
  }>({ clients: [], offres: [] });
  const [loading, setLoading] = useState(false);
  const [notifications, setNotifications] = useState<Notification[]>([]);
  const [notificationsLoading, setNotificationsLoading] = useState(false);
  const [unreadCount, setUnreadCount] = useState(0);
  
  const { toast } = useToast();
  const navigate = useNavigate();
  const user = JSON.parse(localStorage.getItem("user") || '{}');

  // ✅ CORRECTION : Récupérer le rôle depuis l'utilisateur
  const getUserRole = () => {
    // Si l'utilisateur a une propriété 'role'
    if (user.role) {
      return user.role;
    }
    // Sinon, vérifier s'il a des permissions d'admin
    if (user.permissions && Array.isArray(user.permissions)) {
      return user.permissions.includes('admin') ? 'admin' : 'collaborator';
    }
    // Fallback par défaut
    return 'collaborator';
  };

  const userRole = getUserRole();
  
  // ✅ CORRECTION : Afficher le bon libellé selon le rôle
  const getRoleLabel = () => {
    switch (userRole) {
      case 'admin':
        return 'Administrateur';
      case 'collaborator':
        return 'Collaborateur';
      default:
        return 'Utilisateur';
    }
  };

  // ✅ Fonction pour récupérer le token
  const getAuthToken = () => {
    return localStorage.getItem("token");
  };

  // ✅ Fonction pour les appels API avec authentification
  const apiClient = axios.create({
    baseURL: "http://127.0.0.1:8000/api",
    headers: {
      "Content-Type": "application/json",
    },
  });

  // Interceptor pour ajouter le token à chaque requête
  apiClient.interceptors.request.use((config) => {
    const token = getAuthToken();
    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
    }
    return config;
  });

  // ✅ Charger les notifications depuis l'API
  const fetchNotifications = async () => {
    setNotificationsLoading(true);
    try {
      const response = await apiClient.get<Notification[]>("/notifications");
      setNotifications(response.data);
      
      // Calculer le nombre de notifications non lues
      const unread = response.data.filter((notif: Notification) => !notif.read_at).length;
      setUnreadCount(unread);
      
    } catch (error) {
      console.error("Erreur chargement notifications:", error);
      toast({
        title: "Erreur",
        description: "Impossible de charger les notifications",
        variant: "destructive",
      });
    } finally {
      setNotificationsLoading(false);
    }
  };

  // ✅ Charger le compteur de notifications non lues
  const fetchUnreadCount = async () => {
    try {
      const response = await apiClient.get<UnreadCountResponse>("/notifications/unread-count");
      setUnreadCount(response.data.count);
    } catch (error) {
      console.error("Erreur chargement compteur notifications:", error);
    }
  };

  useEffect(() => {
    fetchNotifications();
    
    // Optionnel: Rafraîchir les notifications toutes les 30 secondes
    const interval = setInterval(fetchUnreadCount, 30000);
    return () => clearInterval(interval);
  }, []);

  // ✅ Fonction pour extraire les initiales
  const getInitials = (name: string): string => {
    if (!name) return "U";
    const names = name.trim().split(" ");
    if (names.length === 1) return names[0].charAt(0).toUpperCase();
    return (names[0].charAt(0) + names[names.length - 1].charAt(0)).toUpperCase();
  };

  // ✅ état pour gérer l'ouverture de la popup
  const [openDialog, setOpenDialog] = useState(false);

  const logout = () => {
    localStorage.removeItem("user");
    localStorage.removeItem("token");
    localStorage.removeItem("isAuthenticated");
    window.location.href = "/login";
    navigate("/login");
  };

  // ✅ Fonctions pour les notifications
  const markAsRead = async (notificationId: number) => {
    try {
      await apiClient.put(`/notifications/${notificationId}/read`);
      
      // Mettre à jour l'état local
      setNotifications(notifications.map(notif => 
        notif.id === notificationId ? { ...notif, read_at: new Date().toISOString() } : notif
      ));
      
      // Mettre à jour le compteur
      setUnreadCount(prev => Math.max(0, prev - 1));
      
      toast({
        title: "Notification lue",
        description: t("notificationMarkedReadDesc"),
      });
    } catch (error) {
      console.error("Erreur marquage comme lu:", error);
      toast({
        title: "Erreur",
        description: t("notificationError"),
        variant: "destructive",
      });
    }
  };

  const markAllAsRead = async () => {
    try {
      await apiClient.put("/notifications/mark-all-read");
      
      // Mettre à jour l'état local
      setNotifications(notifications.map(notif => ({ 
        ...notif, 
        read_at: notif.read_at || new Date().toISOString() 
      })));
      
      // Réinitialiser le compteur
      setUnreadCount(0);
      
      toast({
        title: t("allReadTitle"),
        description: t("allReadDesc"),
      });
    } catch (error) {
      console.error("Erreur marquage multiple:", error);
      toast({
        title: t("allReadErrorTitle"),
        description: t("allReadErrorDesc"),
        variant: "destructive",
      });
    }
  };

  // ✅ Fonctions pour formater l'affichage des notifications
  const getNotificationIcon = (type: string) => {
    switch (type) {
      case 'new_client':
        return <Users className="h-4 w-4 text-green-500" />;
      case 'offer_expiring':
        return <Clock className="h-4 w-4 text-red-500" />;
      case 'payment_received':
        return <CreditCard className="h-4 w-4 text-blue-500" />;
      case 'meeting_reminder':
        return <Calendar className="h-4 w-4 text-purple-500" />;
      case 'task_assigned':
        return <FileText className="h-4 w-4 text-orange-500" />;
      case 'system_alert':
        return <AlertTriangle className="h-4 w-4 text-yellow-500" />;
      default:
        return <Bell className="h-4 w-4 text-gray-500" />;
    }
  };

  const getNotificationColor = (type: string) => {
    switch (type) {
      case 'new_client':
        return 'border-l-green-500 bg-green-50/50';
      case 'offer_expiring':
        return 'border-l-red-500 bg-red-50/50';
      case 'payment_received':
        return 'border-l-blue-500 bg-blue-50/50';
      case 'meeting_reminder':
        return 'border-l-purple-500 bg-purple-50/50';
      case 'task_assigned':
        return 'border-l-orange-500 bg-orange-50/50';
      case 'system_alert':
        return 'border-l-yellow-500 bg-yellow-50/50';
      default:
        return 'border-l-gray-500 bg-gray-50/50';
    }
  };

  const getPriorityFromType = (type: string): 'low' | 'medium' | 'high' => {
    switch (type) {
      case 'offer_expiring':
      case 'meeting_reminder':
        return 'high';
      case 'new_client':
      case 'payment_received':
        return 'medium';
      default:
        return 'low';
    }
  };

  const getPriorityBadge = (priority: 'low' | 'medium' | 'high') => {
    switch (priority) {
      case 'high':
        return <Badge variant="destructive" className="h-5 text-xs">{t("priority.high")}</Badge>;
      case 'medium':
        return <Badge variant="secondary" className="h-5 text-xs">{t("priority.medium")}</Badge>;
      case 'low':
        return <Badge variant="outline" className="h-5 text-xs">{t("priority.low")}</Badge>;
      default:
        return null;
    }
  };

  // ✅ Fonction pour formater le temps relatif
  const formatTimeAgo = (dateString: string) => {
    const date = new Date(dateString);
    const now = new Date();
    const diffInSeconds = Math.floor((now.getTime() - date.getTime()) / 1000);
    
    if (diffInSeconds < 60) return 'À l\'instant';
    if (diffInSeconds < 3600) return `Il y a ${Math.floor(diffInSeconds / 60)} min`;
    if (diffInSeconds < 86400) return `Il y a ${Math.floor(diffInSeconds / 3600)} h`;
    if (diffInSeconds < 2592000) return `Il y a ${Math.floor(diffInSeconds / 86400)} j`;
    
    return date.toLocaleDateString('fr-FR');
  };

  // Trier les notifications par date de création et statut de lecture
  const sortedNotifications = [...notifications].sort((a, b) => {
    const aIsUnread = !a.read_at;
    const bIsUnread = !b.read_at;
    
    // Priorité aux non lues
    if (aIsUnread && !bIsUnread) return -1;
    if (!aIsUnread && bIsUnread) return 1;
    
    // Puis par date (plus récent en premier)
    return new Date(b.created_at).getTime() - new Date(a.created_at).getTime();
  });

  // Recherche globale frontend (clients + offres)
  useEffect(() => {
    if (!searchQuery) {
      setSearchResults({ clients: [], offres: [] });
      return;
    }
    setLoading(true);
    
    // Appel API backend pour la recherche globale
    Promise.all([
      fetch(`http://127.0.0.1:8000/api/clients?search=${encodeURIComponent(searchQuery)}`)
        .then(res => res.json())
        .catch(() => []),
      fetch(`http://127.0.0.1:8000/api/offers?search=${encodeURIComponent(searchQuery)}`)
        .then(res => res.json())
        .catch(() => [])
    ]).then(([clients, offres]) => {
      const query = searchQuery.toLowerCase();
      const filteredClients = Array.isArray(clients) ? clients.filter(
        (client: SearchClient) =>
          client.name?.toLowerCase().includes(query) ||
          client.company?.toLowerCase().includes(query) ||
          client.email?.toLowerCase().includes(query)
      ) : [];
      
      const filteredOffres = Array.isArray(offres) ? offres.filter(
        (offre: SearchOffer) =>
          offre.title?.toLowerCase().includes(query) ||
          offre.description?.toLowerCase().includes(query) ||
          offre.category?.toLowerCase().includes(query)
      ) : [];
      
      setSearchResults({ clients: filteredClients, offres: filteredOffres });
      setLoading(false);
    });
  }, [searchQuery]);

  return (
    <header className="h-16 border-b border-border bg-card/50 backdrop-blur-sm sticky top-0 z-40">
      <div className="flex items-center gap-4 px-6 h-full">
        <SidebarTrigger className="lg:hidden" />

        {/* ✅ Barre de recherche */}
        <div className="flex-1 max-w-md relative">
          <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 h-4 w-4 text-muted-foreground" />
          <Input
            placeholder={t("searchPlaceholder")}
            className="pl-10 bg-background border-border focus:border-primary"
            value={searchQuery}
            onChange={(e) => setSearchQuery(e.target.value)}
          />
        </div>

        {/* ✅ Résultats de recherche */}
        {searchQuery && (
          <div className="absolute top-16 left-6 right-6 bg-background border border-border rounded-md shadow-lg max-h-64 overflow-auto z-50">
            {loading ? (
              <div className="p-2 text-muted-foreground">{t("loading")}</div>
            ) : (
              <>
                {searchResults.clients.length > 0 && (
                  <>
                    <div className="p-2 font-semibold text-sm border-b">{t("clients")}</div>
                    {searchResults.clients.map((client) => (
                      <Link
                        key={client.id}
                        to={`/agency/clients/${client.id}`}
                        className="block p-2 hover:bg-muted"
                        onClick={() => setSearchQuery("")}
                      >
                        {client.name} - {client.company}
                      </Link>
                    ))}
                  </>
                )}
                {searchResults.offres.length > 0 && (
                  <>
                    <div className="p-2 font-semibold text-sm border-b">{t("offers")}</div>
                    {searchResults.offres.map((offre) => (
                      <Link
                        key={offre.id}
                        to={`/agency/offres/${offre.id}`}
                        className="block p-2 hover:bg-muted"
                        onClick={() => setSearchQuery("")}
                      >
                        {offre.title} - {offre.client || t("noClient")}
                      </Link>
                    ))}
                  </>
                )}
                {searchResults.clients.length === 0 && searchResults.offres.length === 0 && (
                  <div className="p-2 text-muted-foreground">{t("noResults")}</div>
                )}
              </>
            )}
          </div>
        )}

        {/* ✅ Menu utilisateur */}
        <div className="flex items-center gap-3 ml-auto">
          {/* ✅ Menu des notifications FONCTIONNEL avec API réelle */}
          <DropdownMenu>
            <DropdownMenuTrigger asChild>
              <Button variant="ghost" size="sm" className="relative">
                <Bell className="h-4 w-4" />
                {unreadCount > 0 && (
                  <Badge
                    variant="destructive"
                    className="absolute -top-1 -right-1 h-5 w-5 rounded-full p-0 flex items-center justify-center text-xs"
                  >
                    {unreadCount > 99 ? '99+' : unreadCount}
                  </Badge>
                )}
              </Button>
            </DropdownMenuTrigger>

            <DropdownMenuContent align="end" className="w-96 max-h-96 overflow-auto">
              <DropdownMenuLabel className="flex items-center justify-between">
                <span>{t("notifications")}</span>
                {unreadCount > 0 && (
                  <Button 
                    variant="ghost" 
                    size="sm" 
                    onClick={markAllAsRead}
                    className="h-auto p-0 text-xs text-primary hover:text-primary"
                  >
                    {t("markAllAsRead")}
                  </Button>
                )}
              </DropdownMenuLabel>
              <DropdownMenuSeparator />

              {notificationsLoading ? (
                <div className="p-4 text-center text-muted-foreground">
                  <div className="animate-spin rounded-full h-6 w-6 border-b-2 border-primary mx-auto mb-2"></div>
                  <p className="text-sm">{t("loadingNotifications")}</p>
                </div>
              ) : sortedNotifications.length === 0 ? (
                <div className="p-4 text-center text-muted-foreground">
                  <Bell className="h-8 w-8 mx-auto mb-2 opacity-50" />
                  <p className="text-sm">{t("noNotifications")}</p>
                  <p className="text-xs mt-1">{t("noNotificationsDesc")}</p>
                </div>
              ) : (
                <div className="max-h-64 overflow-auto">
                  {sortedNotifications.map((notification) => {
                    const isUnread = !notification.read_at;
                    const priority = getPriorityFromType(notification.type);
                    
                    return (
                      <DropdownMenuItem 
                        key={notification.id} 
                        className={`p-3 border-l-4 ${getNotificationColor(notification.type)} ${isUnread ? 'bg-muted/30' : ''} hover:bg-muted/50 transition-colors cursor-pointer`}
                        onClick={() => {
                          if (isUnread) {
                            markAsRead(notification.id);
                          }
                          if (notification.action_url) {
                            navigate(notification.action_url);
                          }
                        }}
                      >
                        <div className="flex items-start gap-3 w-full">
                          <div className="flex-shrink-0 mt-0.5">
                            {getNotificationIcon(notification.type)}
                          </div>
                          <div className="flex-1 min-w-0">
                            <div className="flex items-start justify-between gap-2 mb-1">
                              <p className="text-sm font-medium text-foreground flex-1">
                                {notification.title}
                              </p>
                              {getPriorityBadge(priority)}
                            </div>
                            <p className="text-xs text-muted-foreground leading-relaxed">
                              {notification.message}
                            </p>
                            <div className="flex items-center justify-between mt-2">
                              <p className="text-xs text-muted-foreground">
                                {formatTimeAgo(notification.created_at)}
                              </p>
                              {isUnread && (
                                <div className="w-2 h-2 bg-blue-500 rounded-full flex-shrink-0" />
                              )}
                            </div>
                          </div>
                        </div>
                      </DropdownMenuItem>
                    );
                  })}
                </div>
              )}

              <DropdownMenuSeparator />
              <DropdownMenuItem className="justify-center text-primary font-medium text-sm" asChild>
                <Link to="/agency/notifications" className="flex items-center gap-2 w-full justify-center">
                  <Eye className="h-4 w-4" />
                  {t("viewAll")}
                </Link>
              </DropdownMenuItem>
            </DropdownMenuContent>
          </DropdownMenu>

          <DropdownMenu>
            <DropdownMenuTrigger asChild>
              <Button
                variant="ghost"
                className="flex items-center gap-2 hover:bg-primary/5"
              >
                <Avatar className="h-8 w-8">
                  <AvatarFallback className="bg-gradient-primary text-primary-foreground">
                    {getInitials(user?.name || "Utilisateur")}
                  </AvatarFallback>
                </Avatar>
                <div className="hidden md:block text-left">
                  <p className="text-sm font-medium">{user?.name || "Utilisateur"}</p>
                  <p className="text-xs text-muted-foreground"> {t("admin", { ns: "common" })}</p>
                </div>
              </Button>
            </DropdownMenuTrigger>

            <DropdownMenuContent align="end" className="w-56">
              <DropdownMenuLabel>{t("account")}</DropdownMenuLabel>
              <DropdownMenuSeparator />

              <DropdownMenuItem asChild>
                <Link to="/profile" className="flex items-center">
                  <User className="mr-2 h-4 w-4" />
                  {t("myProfile")}
                </Link>
              </DropdownMenuItem>

              <DropdownMenuItem>
                <Settings className="mr-2 h-4 w-4" />
                {t("agencySettings")}
              </DropdownMenuItem>
              <DropdownMenuSeparator />

              <DropdownMenuItem
                className="text-destructive cursor-pointer"
                onClick={() => setOpenDialog(true)}
              >
                <LogOut className="mr-2 h-4 w-4" />
                {t("logout")}
              </DropdownMenuItem>
            </DropdownMenuContent>
          </DropdownMenu>

          {/* ✅ Popup confirmation déconnexion */}
          <AlertDialog open={openDialog} onOpenChange={setOpenDialog}>
            <AlertDialogContent>
              <AlertDialogHeader>
                <AlertDialogTitle>{t("logoutConfirmTitle")}</AlertDialogTitle>
                <AlertDialogDescription>
                {t("logoutConfirmDesc")}
                </AlertDialogDescription>
              </AlertDialogHeader>
              <AlertDialogFooter>
                <AlertDialogCancel onClick={() => setOpenDialog(false)}>
                {t("cancel")}
                </AlertDialogCancel>
                <AlertDialogAction
                  onClick={() => {
                    setOpenDialog(false);
                    logout();
                  }}
                  className="bg-destructive hover:bg-destructive/90"
                >
                  {t("confirmLogout")}
                </AlertDialogAction>
              </AlertDialogFooter>
            </AlertDialogContent>
          </AlertDialog>
        </div>
      </div>
    </header>
  );
}
