// agency-frontend/src/components/agency/EditPermissionsModal.tsx
import { useState, useEffect } from "react";
import { Button } from "@/components/ui/button";
import { Label } from "@/components/ui/label";
import { Checkbox } from "@/components/ui/checkbox";
import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogFooter } from "@/components/ui/dialog";
import { Permission, Collaborator } from "@/types/invitation";
import { updateCollaboratorPermissions } from "@/lib/invitation-api";
import { useToast } from "@/hooks/use-toast";

interface EditPermissionsModalProps {
  open: boolean;
  onOpenChange: (open: boolean) => void;
  collaborator: Collaborator | null;
  onPermissionsUpdated: () => void;
}

// ✅ LISTE COMPLÈTE AVEC TOUTES LES PERMISSIONS
const availablePermissions: { key: Permission; label: string; description: string }[] = [
  {
    key: "view_dashboard",
    label: "Voir le tableau de bord",
    description: "Accéder au tableau de bord principal"
  },
  {
    key: "view_clients", 
    label: "Voir les clients",
    description: "Consulter la liste des clients et leurs profils"
  },
  {
    key: "manage_clients",
    label: "Gérer les clients",
    description: "Créer, modifier et supprimer des clients"
  },
  {
    key: "view_offers",
    label: "Voir les offres",
    description: "Consulter les offres et leurs détails"
  },
  {
    key: "manage_offers",
    label: "Gérer les offres", 
    description: "Créer, modifier et supprimer des offres"
  },
  {
    key: "view_statistics",
    label: "Voir les statistiques",
    description: "Accéder aux statistiques de l'agence"
  },
  // ✅ NOUVELLES PERMISSIONS
  {
    key: "view_settings",
    label: "Voir les paramètres",
    description: "Accéder à la page des paramètres de l'agence"
  },
  {
    key: "manage_agency",
    label: "Gérer l'agence",
    description: "Modifier les informations de l'agence"
  },
  {
    key: "manage_team", 
    label: "Gérer l'équipe",
    description: "Inviter et gérer les collaborateurs"
  }
];

export function EditPermissionsModal({ 
  open, 
  onOpenChange, 
  collaborator, 
  onPermissionsUpdated 
}: EditPermissionsModalProps) {
  const { toast } = useToast();
  const [loading, setLoading] = useState(false);
  const [permissions, setPermissions] = useState<Permission[]>([]);

  useEffect(() => {
    if (collaborator) {
      setPermissions(collaborator.permissions || []);
    }
  }, [collaborator]);

  const handlePermissionChange = (permission: Permission, checked: boolean) => {
    setPermissions(prev => 
      checked ? [...prev, permission] : prev.filter(p => p !== permission)
    );
  };

  const handleSubmit = async () => {
    if (!collaborator) return;

    if (permissions.length === 0) {
      toast({
        title: "Erreur", 
        description: "Au moins une permission doit être sélectionnée",
        variant: "destructive"
      });
      return;
    }

    setLoading(true);
    try {
      console.log("🔄 Mise à jour des permissions...");

      // 1. Mettre à jour côté serveur
      await updateCollaboratorPermissions(collaborator.id, permissions);
      
      // 2. FORCER le rafraîchissement
      const token = localStorage.getItem('token');
      if (token) {
        const response = await fetch('http://127.0.0.1:8000/api/user', {
          headers: { 'Authorization': `Bearer ${token}` }
        });

        if (response.ok) {
          const freshUserData = await response.json();
          localStorage.setItem('user', JSON.stringify(freshUserData));
          window.dispatchEvent(new Event('userPermissionsUpdated'));
        }
      }

      toast({
        title: "✅ Succès", 
        description: "Permissions mises à jour"
      });
      
      onOpenChange(false);
      onPermissionsUpdated();
      
    } catch (error) {
      toast({
        title: "Erreur",
        description: error instanceof Error ? error.message : "Erreur inconnue",
        variant: "destructive"
      });
    } finally {
      setLoading(false);
    }
  };

  // ✅ FONCTION POUR TOUT SÉLECTIONNER
  const selectAll = () => {
    setPermissions(availablePermissions.map(p => p.key));
  };

  // ✅ FONCTION POUR TOUT DÉSÉLECTIONNER
  const deselectAll = () => {
    setPermissions([]);
  };

  if (!collaborator) return null;

  return (
    <Dialog open={open} onOpenChange={onOpenChange}>
      <DialogContent className="sm:max-w-lg max-h-[80vh] overflow-y-auto">
        <DialogHeader>
          <DialogTitle>Modifier les permissions</DialogTitle>
        </DialogHeader>

        <div className="space-y-4">
          <div>
            <p className="text-sm text-muted-foreground">
              Permissions pour <strong>{collaborator.name || collaborator.email}</strong>
            </p>
          </div>

          {/* ✅ BOUTONS SELECTION RAPIDE */}
          <div className="flex gap-2">
            <Button type="button" variant="outline" size="sm" onClick={selectAll}>
              Tout sélectionner
            </Button>
            <Button type="button" variant="outline" size="sm" onClick={deselectAll}>
              Tout désélectionner
            </Button>
          </div>

          <div className="space-y-4">
            <Label>Permissions disponibles</Label>
            <div className="space-y-3 max-h-96 overflow-y-auto p-1">
              {availablePermissions.map((permission) => (
                <div key={permission.key} className="flex items-start space-x-3 p-2 rounded-lg border border-border hover:bg-accent/50 transition-colors">
                  <Checkbox
                    id={`edit-${permission.key}`}
                    checked={permissions.includes(permission.key)}
                    onCheckedChange={(checked) => 
                      handlePermissionChange(permission.key, checked as boolean)
                    }
                    className="mt-1"
                  />
                  <div className="flex-1 min-w-0">
                    <label
                      htmlFor={`edit-${permission.key}`}
                      className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 cursor-pointer block"
                    >
                      {permission.label}
                    </label>
                    <p className="text-xs text-muted-foreground mt-1">
                      {permission.description}
                    </p>
                  </div>
                </div>
              ))}
            </div>
          </div>

          {/* ✅ RÉSUMÉ DES PERMISSIONS SÉLECTIONNÉES */}
          <div className="bg-muted/50 rounded-lg p-3">
            <p className="text-sm font-medium mb-2">
              Permissions sélectionnées: {permissions.length} / {availablePermissions.length}
            </p>
            {permissions.length > 0 ? (
              <div className="flex flex-wrap gap-1">
                {permissions.map(permission => {
                  const permInfo = availablePermissions.find(p => p.key === permission);
                  return (
                    <span key={permission} className="text-xs bg-primary/10 text-primary px-2 py-1 rounded">
                      {permInfo?.label}
                    </span>
                  );
                })}
              </div>
            ) : (
              <p className="text-xs text-muted-foreground">Aucune permission sélectionnée</p>
            )}
          </div>

          <DialogFooter className="flex gap-2 sm:gap-0">
            <Button 
              type="button" 
              variant="outline" 
              onClick={() => onOpenChange(false)}
              disabled={loading}
            >
              Annuler
            </Button>
            <Button onClick={handleSubmit} disabled={loading}>
              {loading ? "Mise à jour..." : "Enregistrer les permissions"}
            </Button>
          </DialogFooter>
        </div>
      </DialogContent>
    </Dialog>
  );
}
