// agency-frontend/src/components/agency/InviteCollaboratorModal.tsx
import { useState } from "react";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import { Checkbox } from "@/components/ui/checkbox";
import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogDescription, DialogFooter } from "@/components/ui/dialog";
import { Permission, InvitationData } from "@/types/invitation";
import { inviteCollaborator } from "@/lib/invitation-api";
import { useToast } from "@/hooks/use-toast";

interface InviteCollaboratorModalProps {
  open: boolean;
  onOpenChange: (open: boolean) => void;
  onInviteSent: () => void;
}

const availablePermissions: { key: Permission; label: string; description: string }[] = [
  {
    key: "view_dashboard",
    label: "Voir le tableau de bord",
    description: "Accéder au tableau de bord principal"
  },
  {
    key: "view_clients", 
    label: "Voir les clients",
    description: "Consulter la liste des clients et leurs profils"
  },
  {
    key: "view_offers",
    label: "Voir les offres",
    description: "Consulter les offres et leurs détails"
  },
  {
    key: "view_statistics",
    label: "Voir les statistiques",
    description: "Accéder aux statistiques de l'agence"
  }
];

export function InviteCollaboratorModal({ open, onOpenChange, onInviteSent }: InviteCollaboratorModalProps) {
  const { toast } = useToast();
  const [loading, setLoading] = useState(false);
  const [formData, setFormData] = useState<InvitationData>({
    name: "",
    email: "",
    permissions: ["view_dashboard"]
  });

  const handlePermissionChange = (permission: Permission, checked: boolean) => {
    setFormData(prev => ({
      ...prev,
      permissions: checked 
        ? [...prev.permissions, permission]
        : prev.permissions.filter(p => p !== permission)
    }));
  };

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    
    if (!formData.email) {
      toast({
        title: "Erreur",
        description: "L'email est obligatoire",
        variant: "destructive"
      });
      return;
    }

    if (formData.permissions.length === 0) {
      toast({
        title: "Erreur", 
        description: "Au moins une permission doit être sélectionnée",
        variant: "destructive"
      });
      return;
    }

    setLoading(true);
    try {
      await inviteCollaborator(formData);
      toast({
        title: "Succès",
        description: "Invitation envoyée avec succès"
      });
      setFormData({ name: "", email: "", permissions: ["view_dashboard"] });
      onOpenChange(false);
      onInviteSent();
    } catch (error) {
      toast({
        title: "Erreur",
        description: error instanceof Error ? error.message : "Erreur inconnue",
        variant: "destructive"
      });
    } finally {
      setLoading(false);
    }
  };

  return (
    <Dialog open={open} onOpenChange={onOpenChange}>
      <DialogContent className="sm:max-w-md">
        <DialogHeader>
          <DialogTitle>Inviter un collaborateur</DialogTitle>
          <DialogDescription>
            Envoyer une invitation par email à un nouveau collaborateur
          </DialogDescription>
        </DialogHeader>

        <form onSubmit={handleSubmit} className="space-y-4">
          {/* Champ Nom (optionnel) */}
          <div className="space-y-2">
            <Label htmlFor="name">Nom du collaborateur (optionnel)</Label>
            <Input
              id="name"
              placeholder="Ex: Jean Dupont"
              value={formData.name}
              onChange={(e) => setFormData(prev => ({ ...prev, name: e.target.value }))}
            />
            <p className="text-xs text-muted-foreground">
              Si aucun nom n'est saisi, l'email sera utilisé pour l'affichage
            </p>
          </div>

          {/* Email */}
          <div className="space-y-2">
            <Label htmlFor="email">Email du collaborateur *</Label>
            <Input
              id="email"
              type="email"
              placeholder="collaborateur@exemple.com"
              value={formData.email}
              onChange={(e) => setFormData(prev => ({ ...prev, email: e.target.value }))}
              required
            />
          </div>

          {/* Permissions */}
          <div className="space-y-3">
            <Label>Permissions *</Label>
            <div className="space-y-2">
              {availablePermissions.map((permission) => (
                <div key={permission.key} className="flex items-start space-x-2">
                  <Checkbox
                    id={permission.key}
                    checked={formData.permissions.includes(permission.key)}
                    onCheckedChange={(checked) => 
                      handlePermissionChange(permission.key, checked as boolean)
                    }
                  />
                  <div className="grid gap-1.5 leading-none">
                    <label
                      htmlFor={permission.key}
                      className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                    >
                      {permission.label}
                    </label>
                    <p className="text-sm text-muted-foreground">
                      {permission.description}
                    </p>
                  </div>
                </div>
              ))}
            </div>
          </div>

          <DialogFooter>
            <Button 
              type="button" 
              variant="outline" 
              onClick={() => onOpenChange(false)}
              disabled={loading}
            >
              Annuler
            </Button>
            <Button type="submit" disabled={loading}>
              {loading ? "Envoi en cours..." : "Envoyer l'invitation"}
            </Button>
          </DialogFooter>
        </form>
      </DialogContent>
    </Dialog>
  );
}
