// agency-frontend/src/pages/Login.tsx
import { useState, useEffect } from "react";
import { useNavigate, Link, useSearchParams } from "react-router-dom";
import { apiUrl } from "@/lib/config";
import { User, Lock, Eye, EyeOff, Loader2, Sparkles, CheckCircle } from "lucide-react";

type LoginResponse = {
  access_token: string;
  token_type: string;
  user?: {
    id: number;
    name: string;
    email: string;
  };
};

export default function Login() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [showPassword, setShowPassword] = useState(false);
  const [loading, setLoading] = useState(false);
  const [message, setMessage] = useState("");
  const [error, setError] = useState("");
  const [searchParams] = useSearchParams();
  const navigate = useNavigate();

  // ✅ GESTION DES INVITATIONS
  useEffect(() => {
    const invitationAccepted = searchParams.get('invitation_accepted');
    const invitationEmail = searchParams.get('email');
    const errorType = searchParams.get('error');

    if (invitationAccepted && invitationEmail) {
      setEmail(invitationEmail);
      setMessage("🎉 Invitation acceptée ! Vous pouvez maintenant vous connecter.");
    }

    if (errorType === 'invalid_invitation') {
      setError("❌ Lien d'invitation invalide ou expiré");
    }

    if (errorType === 'invitation_already_used') {
      setError("⚠️ Cette invitation a déjà été utilisée");
    }
  }, [searchParams]);

  const passwordCriteria = [
    { label: "Au moins 8 caractères", test: (pw: string) => pw.length >= 8 },
    { label: "Au moins une majuscule", test: (pw: string) => /[A-Z]/.test(pw) },
    { label: "Au moins une minuscule", test: (pw: string) => /[a-z]/.test(pw) },
    { label: "Au moins un chiffre", test: (pw: string) => /[0-9]/.test(pw) },
  ];

  const unmetCriteria = passwordCriteria.filter((c) => !c.test(password));

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    setLoading(true);
    setMessage("");
    setError("");

    try {
      const res = await fetch(apiUrl("/login"), {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ email, password }),
      });

      if (!res.ok) {
        if (res.status === 429) {
          setError("Trop de tentatives. Réessayez dans une minute.");
        } else {
          setError("Identifiants incorrects. Veuillez réessayer.");
        }
        return;
      }

      const data: LoginResponse = await res.json();
      localStorage.setItem("token", data.access_token);
      localStorage.setItem("user", JSON.stringify(data.user));

      setMessage("Connexion réussie !");
      setTimeout(() => navigate("/dashboard"), 1000);
    } catch (err: unknown) {
      if (err instanceof Error) setError(err.message);
      else setError("Une erreur est survenue. Veuillez réessayer.");
    } finally {
      setLoading(false);
    }
  };

  return (
    <div className="min-h-screen bg-gradient-to-br from-gray-50 to-gray-100 flex items-center justify-center p-4">
      <div className="relative w-full max-w-md">
        <form
          onSubmit={handleSubmit}
          className="bg-white rounded-2xl shadow-lg p-8 border border-gray-200"
        >
          {/* En-tête */}
          <div className="text-center mb-8">
            <div className="flex justify-center mb-4">
              <div className="relative">
                <div className="w-16 h-16 bg-gradient-to-br from-blue-500 to-purple-600 rounded-2xl flex items-center justify-center shadow-lg">
                  <svg 
                    className="w-8 h-8 text-white" 
                    fill="none" 
                    stroke="currentColor" 
                    viewBox="0 0 24 24"
                  >
                    <path 
                      strokeLinecap="round" 
                      strokeLinejoin="round" 
                      strokeWidth={2} 
                      d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z" 
                    />
                  </svg>
                </div>
                <div className="absolute -top-1 -right-1 w-6 h-6 bg-yellow-400 rounded-full flex items-center justify-center">
                  <Sparkles className="w-3 h-3 text-white" />
                </div>
              </div>
            </div>
            
            <h1 className="text-3xl font-bold text-gray-900 mb-2">
              Content de vous revoir
            </h1>
            <p className="text-gray-600">Connectez-vous à votre espace personnel</p>
          </div>

          {/* Messages */}
          {message && (
            <div className="mb-6 p-4 bg-green-50 border border-green-200 rounded-lg text-green-700 text-center flex items-center justify-center gap-2">
              <CheckCircle className="w-5 h-5" />
              {message}
            </div>
          )}
          {error && (
            <div className="mb-6 p-4 bg-red-50 border border-red-200 rounded-lg text-red-700 text-center">
              {error}
            </div>
          )}

          {/* Champ email */}
          <div className="mb-5">
            <label className="block text-sm font-medium text-gray-700 mb-3">
              Email
            </label>
            <div className="relative">
              <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                <User className="h-5 w-5 text-gray-400" />
              </div>
              <input
                type="email"
                value={email}
                onChange={(e) => setEmail(e.target.value)}
                className="block w-full pl-11 pr-4 py-3 bg-white border border-gray-300 rounded-lg focus:ring-2 focus:ring-gray-400 focus:border-gray-400 transition-all duration-200 outline-none placeholder-gray-400"
                placeholder="votre@email.com"
                required
              />
            </div>
          </div>

          {/* Champ mot de passe */}
          <div className="mb-4">
            <label className="block text-sm font-medium text-gray-700 mb-3">
              Mot de passe
            </label>
            <div className="relative">
              <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                <Lock className="h-5 w-5 text-gray-400" />
              </div>
              <input
                type={showPassword ? "text" : "password"}
                value={password}
                onChange={(e) => setPassword(e.target.value)}
                className="block w-full pl-11 pr-12 py-3 bg-white border border-gray-300 rounded-lg focus:ring-2 focus:ring-gray-400 focus:border-gray-400 transition-all duration-200 outline-none placeholder-gray-400"
                placeholder="Votre mot de passe"
                required
              />
              <button
                type="button"
                onClick={() => setShowPassword(!showPassword)}
                className="absolute inset-y-0 right-0 pr-3 flex items-center text-gray-400 hover:text-gray-600 transition-colors"
              >
                {showPassword ? <EyeOff className="h-5 w-5" /> : <Eye className="h-5 w-5" />}
              </button>
            </div>
          </div>

          {/* Critères de mot de passe */}
          {password && unmetCriteria.length > 0 && (
            <div className="mb-6 p-4 bg-yellow-50 border border-yellow-200 rounded-lg">
              <p className="text-sm font-medium text-yellow-800 mb-2">
                Critères à respecter :
              </p>
              <ul className="text-sm text-yellow-700 space-y-1.5">
                {unmetCriteria.map((c, i) => (
                  <li key={i} className="flex items-center">
                    <div className="w-1.5 h-1.5 bg-yellow-500 rounded-full mr-3"></div>
                    <span>{c.label}</span>
                  </li>
                ))}
              </ul>
            </div>
          )}

          {/* Bouton de connexion */}
          <button
            type="submit"
            disabled={loading}
            className="w-full bg-gray-900 hover:bg-gray-800 text-white py-3.5 px-4 rounded-lg font-medium shadow-md hover:shadow-lg transition-all duration-200 disabled:opacity-50 flex items-center justify-center"
          >
            {loading ? (
              <>
                <Loader2 className="animate-spin h-5 w-5 mr-2" />
                <span>Connexion...</span>
              </>
            ) : (
              "Se connecter"
            )}
          </button>

          {/* Liens */}
          <div className="flex justify-between mt-6 text-sm">
            <Link
              to="/forgot-password"
              className="text-gray-600 hover:text-gray-800 transition-colors font-medium hover:underline"
            >
              Mot de passe oublié ?
            </Link>
            <Link
              to="/register"
              className="text-gray-600 hover:text-gray-800 transition-colors font-medium hover:underline"
            >
              Créer un compte
            </Link>
          </div>
        </form>
      </div>
    </div>
  );
}
