// agency-frontend/src/pages/NotificationsPage.tsx
// src/pages/agency/NotificationsPage.tsx
import { useState, useEffect } from "react";
import { Bell, ArrowLeft, Users, Clock, CreditCard, Calendar, FileText, AlertTriangle } from "lucide-react";
import { Button } from "@/components/ui/button";
import { Badge } from "@/components/ui/badge";
import { useNavigate } from "react-router-dom";
import { useToast } from "@/hooks/use-toast";
import { useTranslation } from 'react-i18next';

interface Notification {
  id: number;
  user_id: number;
  type: 'new_client' | 'offer_expiring' | 'payment_received' | 'meeting_reminder' | 'task_assigned' | 'system_alert';
  title: string;
  message: string;
  data: any;
  action_url: string | null;
  read_at: string | null;
  created_at: string;
  updated_at: string;
}

export default function NotificationsPage() {
  const { t } = useTranslation("notification");
  const [notifications, setNotifications] = useState<Notification[]>([]);
  const [loading, setLoading] = useState(true);
  const navigate = useNavigate();
  const { toast } = useToast();

  useEffect(() => {
    fetchNotifications();
  }, []);

  const fetchNotifications = async () => {
    try {
      const token = localStorage.getItem("token");
      const response = await fetch("http://127.0.0.1:8000/api/notifications", {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      const data = await response.json();
      setNotifications(data);
    } catch (error) {
      console.error("Erreur:", error);
      toast({
        title: t("error"),
        description: t("errorLoad"),
        variant: "destructive",
      });
    } finally {
      setLoading(false);
    }
  };

  const markAsRead = async (notificationId: number) => {
    try {
      const token = localStorage.getItem("token");
      await fetch(`http://127.0.0.1:8000/api/notifications/${notificationId}/read`, {
        method: "PUT",
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      fetchNotifications(); // Recharger
    } catch (error) {
      console.error("Erreur:", error);
    }
  };

  const markAllAsRead = async () => {
    try {
      const token = localStorage.getItem("token");
      await fetch("http://127.0.0.1:8000/api/notifications/mark-all-read", {
        method: "PUT",
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      fetchNotifications(); // Recharger
      toast({
        title: t("successMarkAll"),
        description: t("successMarkAllDesc"),
      });
    } catch (error) {
      console.error("Erreur:", error);
    }
  };

  const getNotificationIcon = (type: string) => {
    switch (type) {
      case 'new_client': return <Users className="h-5 w-5 text-green-500" />;
      case 'offer_expiring': return <Clock className="h-5 w-5 text-red-500" />;
      case 'payment_received': return <CreditCard className="h-5 w-5 text-blue-500" />;
      case 'meeting_reminder': return <Calendar className="h-5 w-5 text-purple-500" />;
      case 'task_assigned': return <FileText className="h-5 w-5 text-orange-500" />;
      case 'system_alert': return <AlertTriangle className="h-5 w-5 text-yellow-500" />;
      default: return <Bell className="h-5 w-5 text-gray-500" />;
    }
  };

  const getNotificationColor = (type: string) => {
    switch (type) {
      case 'new_client': return 'border-l-green-500 bg-green-50';
      case 'offer_expiring': return 'border-l-red-500 bg-red-50';
      case 'payment_received': return 'border-l-blue-500 bg-blue-50';
      case 'meeting_reminder': return 'border-l-purple-500 bg-purple-50';
      case 'task_assigned': return 'border-l-orange-500 bg-orange-50';
      case 'system_alert': return 'border-l-yellow-500 bg-yellow-50';
      default: return 'border-l-gray-500 bg-gray-50';
    }
  };

  const unreadCount = notifications.filter(n => !n.read_at).length;

  if (loading) {
    return (
      <div className="p-6">
        <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-primary mx-auto"></div>
        <p className="text-center mt-2">{t("loading")}</p>
      </div>
    );
  }

  return (
    <div className="container mx-auto p-6">
      <div className="flex items-center gap-4 mb-6">
        <Button variant="ghost" onClick={() => navigate("/agency")}>
          <ArrowLeft className="h-4 w-4" />
        </Button>
        <div>
          <h1 className="text-2xl font-bold">{t("allNotifications")}</h1>
          <p className="text-sm text-gray-500">
          {t("count", { total: notifications.length, unread: unreadCount })}
          </p>
        </div>
        
        {unreadCount > 0 && (
          <Button onClick={markAllAsRead} className="ml-auto">
            {t("markAllAsRead")}
          </Button>
        )}
      </div>

      <div className="space-y-4">
        {notifications.length === 0 ? (
          <div className="text-center p-8 border rounded-lg">
            <Bell className="h-12 w-12 mx-auto text-gray-400 mb-4" />
            <p className="text-gray-500 text-lg">{t("noNotifications")}</p>
            <p className="text-sm text-gray-400 mt-2">
            {t("noNotificationsDesc")}
            </p>
          </div>
        ) : (
          notifications.map((notification) => (
          
            <div
              key={notification.id}
              className={`p-4 border rounded-lg border-l-4 ${getNotificationColor(notification.type)} ${
                !notification.read_at ? 'ring-2 ring-blue-200' : ''
              }`}
            >
              <div className="flex items-start gap-3">
                <div className="flex-shrink-0 mt-1">
                  {getNotificationIcon(notification.type)}
                </div>
                <div className="flex-1">
                  <div className="flex items-start justify-between">
                    <h3 className="font-semibold text-gray-900">
                    {/* {notification.title} */}
                    {t("newClientTitle")}
                    </h3>
                    {!notification.read_at && (
                      <Badge variant="secondary" className="text-xs">
                        {t("unread")}
                      </Badge>
                    )}
                  </div>
                  <p className="text-sm text-gray-600 mt-1">
                  {notification.message}
                  {/* {t("newClientMessage", { client: notification.data?.client })} */}
                  </p>
                  <div className="flex items-center justify-between mt-3">
                    <p className="text-xs text-gray-400">
                      {new Date(notification.created_at).toLocaleDateString('fr-FR', {
                        day: 'numeric',
                        month: 'long',
                        year: 'numeric',
                        hour: '2-digit',
                        minute: '2-digit'
                      })}
                    </p>
                    <div className="flex gap-2">
                      {!notification.read_at && (
                        <Button 
                          variant="outline" 
                          size="sm"
                          onClick={() => markAsRead(notification.id)}
                        >
                          {t("markAsRead")}
                        </Button>
                      )}
                      {notification.action_url && (
                        <Button 
                          variant="default" 
                          size="sm"
                          onClick={() => navigate(notification.action_url || '/agency')}
                        >
                          {t("view")}
                        </Button>
                      )}
                    </div>
                  </div>
                </div>
              </div>
            </div>
          ))
        )}
      </div>
    </div>
  );
}
