// agency-frontend/src/pages/Register.tsx
import { useState, useEffect } from "react";
import axios from "axios";
import { useAuth } from "../context/AuthContext";
import { Eye, EyeOff, UserPlus, User, Lock, MailCheck } from "lucide-react";
import { Link, useNavigate } from "react-router-dom";

// ✅ Typage de la réponse API Laravel
type RegisterResponse = {
  message?: string;
  error?: string;
  user?: any;
  token?: string; // 🔴 AJOUT: Token Sanctum
};

type ResendResponse = {
  status?: string;
  message?: string;
};

// ✅ Type guard pour erreurs Axios
function isAxiosError<T>(error: unknown): error is { response?: { data?: T } } {
  return typeof error === "object" && error !== null && "response" in error;
}

const Register: React.FC = () => {
  const { setAuthUser } = useAuth();
  const navigate = useNavigate();

  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [passwordConfirmation, setPasswordConfirmation] = useState("");
  const [acceptedTerms, setAcceptedTerms] = useState(false);

  const [error, setError] = useState("");
  const [success, setSuccess] = useState("");
  const [isLoading, setIsLoading] = useState(false);

  const [showPassword, setShowPassword] = useState(false);

  const [isSubmitted, setIsSubmitted] = useState(false);
  const [canResend, setCanResend] = useState(false);
  const [countdown, setCountdown] = useState(0);

  const [passwordCriteria, setPasswordCriteria] = useState({
    length: false,
    uppercase: false,
    lowercase: false,
    number: false,
  });
  const [showPasswordCriteria, setShowPasswordCriteria] = useState(false);

  // ✅ Gestion de l'invitation
  const [invitationToken, setInvitationToken] = useState<string | null>(null);
  const [isInvitation, setIsInvitation] = useState(false);

  // 🔁 Compteur pour renvoi d'email
  useEffect(() => {
    let timer: NodeJS.Timeout;
    if (countdown > 0) {
      timer = setTimeout(() => setCountdown((c) => c - 1), 1000);
    } else {
      setCanResend(true);
    }
    return () => clearTimeout(timer);
  }, [countdown]);

  // ✅ Récupération du token d'invitation
  useEffect(() => {
    const urlParams = new URLSearchParams(window.location.search);
    const token = urlParams.get('invitation_token');
    const invitationEmail = urlParams.get('email');
    const invitationAccepted = urlParams.get('invitation_accepted');
    
    if (token) {
      setInvitationToken(token);
      localStorage.setItem('invitation_token', token);
      setIsInvitation(true);
    }
    
    if (invitationEmail) {
      setEmail(invitationEmail);
    }
    
    if (invitationAccepted === 'true') {
      setSuccess("✅ Invitation acceptée ! Complétez votre inscription.");
    }
  }, []);

  // ✅ Validation
  const validateName = (val: string) => (!val ? "Le nom est requis" : "");
  const validateEmail = (val: string) =>
    /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(val) ? "" : "Email invalide";
  const validatePassword = (val: string) => {
    const hasLength = val.length >= 8;
    const hasUpper = /[A-Z]/.test(val);
    const hasLower = /[a-z]/.test(val);
    const hasNumber = /[0-9]/.test(val);

    setPasswordCriteria({
      length: hasLength,
      uppercase: hasUpper,
      lowercase: hasLower,
      number: hasNumber,
    });

    if (!hasLength) return "Au moins 8 caractères";
    if (!hasUpper) return "Au moins 1 majuscule";
    if (!hasLower) return "Au moins 1 minuscule";
    if (!hasNumber) return "Au moins 1 chiffre";
    return "";
  };

  // Handlers
  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setName(e.target.value.replace(/^\w/, (c) => c.toUpperCase()));
  };

  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setPassword(e.target.value);
    validatePassword(e.target.value);
  };

  const handleConfirmPasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setPasswordConfirmation(e.target.value);
  };

  const handlePasswordFocus = () => setShowPasswordCriteria(true);
  const handlePasswordBlur = () => {
    if (!password) setShowPasswordCriteria(false);
  };

  // 🚀 Register - VERSION CORRIGÉE
  const handleRegister = async (e: React.FormEvent) => {
    e.preventDefault();
    const nameErr = validateName(name);
    const emailErr = validateEmail(email);
    const passwordErr = validatePassword(password);

    if (nameErr || emailErr || passwordErr) {
      setError(nameErr || emailErr || passwordErr);
      return;
    }

    if (password !== passwordConfirmation) {
      setError("Les mots de passe ne correspondent pas");
      return;
    }

    if (!acceptedTerms) {
      setError("Vous devez accepter les conditions d'utilisation");
      return;
    }

    setIsLoading(true);
    setError("");
    setSuccess("");

    try {
      const payload: any = { 
        name, 
        email, 
        password, 
        password_confirmation: passwordConfirmation 
      };

      if (invitationToken) {
        payload.invitation_token = invitationToken;
      }

      const response = await axios.post<RegisterResponse>(
        "http://localhost:8000/api/register",
        payload,
        { withCredentials: true }
      );

      // ✅ SUCCÈS: Gérer la réponse avec token
      if (invitationToken) {
        // 🔴 CORRECTION: Utiliser ton AuthContext pour stocker le token et l'utilisateur
        if (response.data.user && response.data.token) {
          // Stocker le token dans le localStorage (comme dans ton AuthContext)
          localStorage.setItem("token", response.data.token);
          localStorage.setItem("user", JSON.stringify(response.data.user));
          
          // Mettre à jour le contexte d'authentification
          setAuthUser(response.data.user);
          
          setSuccess("🎉 Compte créé avec succès ! Connexion automatique...");
          localStorage.removeItem('invitation_token');
          
          // Redirection vers le dashboard
          setTimeout(() => {
            navigate('/dashboard');
          }, 1000);
        } else {
          // Fallback si pas de token dans la réponse
          setSuccess("🎉 Compte créé ! Redirection vers la connexion...");
          setTimeout(() => {
            navigate('/login');
          }, 1000);
        }
      } else {
        // Comportement normal pour inscription sans invitation
        setIsSubmitted(true);
        setSuccess("Inscription réussie ! Vérifiez votre email.");
        setCountdown(60);
        setCanResend(false);
      }
    } catch (err: unknown) {
      if (isAxiosError<RegisterResponse>(err)) {
        setError(err.response?.data?.message || err.response?.data?.error || "Erreur lors de l'inscription");
      } else {
        setError("Erreur inattendue");
      }
    } finally {
      setIsLoading(false);
    }
  };

  // 📧 Resend email
  const handleResendEmail = async () => {
    setIsLoading(true);
    setError("");
    setSuccess("");

    try {
      const res = await axios.post<ResendResponse>(
        "http://localhost:8000/api/email/verification-notification",
        { email }
      );
      const status = res.data.status ?? res.data.message ?? "verification_link_sent";

      if (status === "already_verified") setSuccess("Votre email est déjà vérifié.");
      else setSuccess("Lien de vérification renvoyé ! Vérifiez votre boîte.");

      setCountdown(60);
      setCanResend(false);
      setIsSubmitted(true);
    } catch (err: unknown) {
      if (isAxiosError<ResendResponse>(err)) setError(err.response?.data?.message || "Impossible de renvoyer l'email.");
      else setError("Erreur inattendue");
    } finally {
      setIsLoading(false);
    }
  };

  // ✅ JSX
  return (
    <div className="min-h-screen bg-gradient-to-br from-gray-50 to-gray-100 flex items-center justify-center p-4">
      <div className="relative w-full max-w-md">
        <form
          onSubmit={handleRegister}
          className="bg-white rounded-2xl shadow-lg p-8 border border-gray-200"
        >
          {/* En-tête - MODIFIÉ pour les invitations */}
          <div className="text-center mb-8">
            {isInvitation ? (
              <div className="flex items-center justify-center mb-4">
                <MailCheck className="h-8 w-8 text-green-600 mr-2" />
                <h1 className="text-2xl font-bold text-gray-900">
                  Accepter l'invitation
                </h1>
              </div>
            ) : (
              <h1 className="text-2xl font-bold text-gray-900 mb-2">
                Créer un compte
              </h1>
            )}
            <p className="text-gray-600 text-sm">
              {isInvitation 
                ? "Complétez votre inscription pour rejoindre l'agence" 
                : "Rejoignez notre communauté"
              }
            </p>
          </div>

          {/* Messages */}
          {error && (
            <div className="mb-6 p-4 bg-red-50 border border-red-200 rounded-lg text-red-700 text-center">
              {error}
            </div>
          )}
          {success && (
            <div className="mb-6 p-4 bg-green-50 border border-green-200 rounded-lg text-green-700 text-center">
              {success}
            </div>
          )}

          {/* Champ nom */}
          <div className="mb-5">
            <label className="block text-sm font-medium text-gray-700 mb-3">
              Nom
            </label>
            <div className="relative">
              <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                <User className="h-5 w-5 text-gray-400" />
              </div>
              <input
                type="text"
                value={name}
                onChange={handleNameChange}
                className="block w-full pl-11 pr-4 py-3 bg-white border border-gray-300 rounded-lg focus:ring-2 focus:ring-gray-400 focus:border-gray-400 transition-all duration-200 outline-none placeholder-gray-400"
                placeholder="Votre nom complet"
                required
              />
            </div>
          </div>

          {/* Champ email */}
          <div className="mb-5">
            <label className="block text-sm font-medium text-gray-700 mb-3">
              Email
            </label>
            <div className="relative">
              <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                <UserPlus className="h-5 w-5 text-gray-400" />
              </div>
              <input
                type="email"
                value={email}
                onChange={(e) => setEmail(e.target.value)}
                className={`block w-full pl-11 pr-4 py-3 bg-white border border-gray-300 rounded-lg focus:ring-2 focus:ring-gray-400 focus:border-gray-400 transition-all duration-200 outline-none placeholder-gray-400 ${
                  isInvitation ? 'bg-gray-50' : ''
                }`}
                placeholder="votre@email.com"
                required
                disabled={isInvitation}
              />
            </div>
            {isInvitation && (
              <p className="text-xs text-gray-500 mt-1">
                Email verrouillé - défini par l'invitation
              </p>
            )}
          </div>

          {/* Champ mot de passe */}
          <div className="mb-5">
            <label className="block text-sm font-medium text-gray-700 mb-3">
              Mot de passe
            </label>
            <div className="relative">
              <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                <Lock className="h-5 w-5 text-gray-400" />
              </div>
              <input
                type={showPassword ? "text" : "password"}
                value={password}
                onChange={handlePasswordChange}
                onFocus={handlePasswordFocus}
                onBlur={handlePasswordBlur}
                className="block w-full pl-11 pr-12 py-3 bg-white border border-gray-300 rounded-lg focus:ring-2 focus:ring-gray-400 focus:border-gray-400 transition-all duration-200 outline-none placeholder-gray-400"
                placeholder="Votre mot de passe"
                required
              />
              <button
                type="button"
                onClick={() => setShowPassword(!showPassword)}
                className="absolute inset-y-0 right-0 pr-3 flex items-center text-gray-400 hover:text-gray-600 transition-colors"
              >
                {showPassword ? <EyeOff className="h-5 w-5" /> : <Eye className="h-5 w-5" />}
              </button>
            </div>

            {/* Critères de mot de passe */}
            {showPasswordCriteria && (
              <div className="mt-3 p-4 bg-gray-50 border border-gray-200 rounded-lg space-y-2">
                <div className={`text-sm flex items-center ${passwordCriteria.length ? "text-green-600" : "text-gray-500"}`}>
                  <span className={`w-2 h-2 rounded-full mr-3 ${passwordCriteria.length ? "bg-green-500" : "bg-gray-300"}`}></span>
                  Au moins 8 caractères
                </div>
                <div className={`text-sm flex items-center ${passwordCriteria.uppercase ? "text-green-600" : "text-gray-500"}`}>
                  <span className={`w-2 h-2 rounded-full mr-3 ${passwordCriteria.uppercase ? "bg-green-500" : "bg-gray-300"}`}></span>
                  Au moins 1 majuscule
                </div>
                <div className={`text-sm flex items-center ${passwordCriteria.lowercase ? "text-green-600" : "text-gray-500"}`}>
                  <span className={`w-2 h-2 rounded-full mr-3 ${passwordCriteria.lowercase ? "bg-green-500" : "bg-gray-300"}`}></span>
                  Au moins 1 minuscule
                </div>
                <div className={`text-sm flex items-center ${passwordCriteria.number ? "text-green-600" : "text-gray-500"}`}>
                  <span className={`w-2 h-2 rounded-full mr-3 ${passwordCriteria.number ? "bg-green-500" : "bg-gray-300"}`}></span>
                  Au moins 1 chiffre
                </div>
              </div>
            )}
          </div>

          {/* Confirmation mot de passe */}
          <div className="mb-6">
            <label className="block text-sm font-medium text-gray-700 mb-3">
              Confirmer le mot de passe
            </label>
            <div className="relative">
              <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                <Lock className="h-5 w-5 text-gray-400" />
              </div>
              <input
                type={showPassword ? "text" : "password"}
                value={passwordConfirmation}
                onChange={handleConfirmPasswordChange}
                className="block w-full pl-11 pr-12 py-3 bg-white border border-gray-300 rounded-lg focus:ring-2 focus:ring-gray-400 focus:border-gray-400 transition-all duration-200 outline-none placeholder-gray-400"
                placeholder="Confirmez votre mot de passe"
                required
              />
              <button
                type="button"
                onClick={() => setShowPassword(!showPassword)}
                className="absolute inset-y-0 right-0 pr-3 flex items-center text-gray-400 hover:text-gray-600 transition-colors"
              >
                {showPassword ? <EyeOff className="h-5 w-5" /> : <Eye className="h-5 w-5" />}
              </button>
            </div>
          </div>

          {/* Conditions d'utilisation */}
          <div className="mb-6">
            <div className="flex items-start space-x-3">
              <input
                type="checkbox"
                id="terms"
                checked={acceptedTerms}
                onChange={(e) => setAcceptedTerms(e.target.checked)}
                className="mt-0.5 w-4 h-4 text-gray-700 bg-gray-100 border-gray-300 rounded focus:ring-gray-500 focus:ring-2"
                required
              />
              <label htmlFor="terms" className="text-sm text-gray-700">
                J'accepte les{" "}
                <Link 
                  to="/conditions-utilisation" 
                  className="text-gray-700 hover:text-gray-900 font-medium underline"
                >
                  conditions d'utilisation
                </Link>{" "}
                et la{" "}
                <Link 
                  to="/politique-confidentialite" 
                  className="text-gray-700 hover:text-gray-900 font-medium underline"
                >
                  politique de confidentialité
                </Link>
              </label>
            </div>
          </div>

          {/* Bouton d'inscription - MODIFIÉ */}
          <button
            type="submit"
            disabled={isLoading}
            className="w-full bg-gray-900 hover:bg-gray-800 text-white py-3.5 px-4 rounded-lg font-medium shadow-md hover:shadow-lg transition-all duration-200 disabled:opacity-50 flex items-center justify-center"
          >
            {isLoading 
              ? "Inscription..." 
              : isInvitation 
                ? "🎉 Rejoindre l'agence" 
                : "S'inscrire"
            }
          </button>

          {/* Renvoyer l'email - CACHÉ pour les invitations */}
          {isSubmitted && !isInvitation && (
            <div className="mt-6 text-center">
              {!canResend ? (
                <p className="text-sm text-gray-500">Vous pourrez renvoyer l'email après {countdown}s</p>
              ) : (
                <button
                  onClick={handleResendEmail}
                  disabled={isLoading}
                  className="w-full bg-gray-700 hover:bg-gray-600 text-white font-medium py-2.5 rounded-lg disabled:opacity-50 transition-all duration-200"
                >
                  {isLoading ? "Envoi en cours..." : "Renvoyer l'email"}
                </button>
              )}
            </div>
          )}

          {/* Lien de connexion - CACHÉ pour les invitations */}
          {!isInvitation && (
            <div className="mt-6 text-center">
              <p className="text-sm text-gray-600">
                Déjà un compte ?{" "}
                <Link to="/login" className="text-gray-700 hover:text-gray-900 font-medium underline">
                  Se connecter
                </Link>
              </p>
            </div>
          )}
        </form>
      </div>
    </div>
  );
};

export default Register;
