// agency-frontend/src/i18n.ts
// src/i18n.ts
import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import LanguageDetector from "i18next-browser-languagedetector";

// FR imports
import commonFr from "./locales/fr/common.json";
import dashboardFr from "./locales/fr/dashboard.json";
import offersFr from "./locales/fr/offers.json";
import clientsFr from "./locales/fr/clients.json";
import settingsFr from "./locales/fr/settings.json";
import statisticsFr from "./locales/fr/statistics.json";
import ClientListFr from "./locales/fr/ClientList.json";
import OfferProfilFr from "./locales/fr/OfferProfil.json";
import AgencyHeaderFr from "./locales/fr/AgencyHeader.json";
import notificationFr from "./locales/fr/notification.json";
import profilFr from "./locales/fr/profil.json";

//import profileFr from "./locales/fr/profile.json";

// EN imports
import commonEn from "./locales/en/common.json";
import dashboardEn from "./locales/en/dashboard.json";
import offersEn from "./locales/en/offers.json";
import clientsEn from "./locales/en/clients.json";
import settingsEn from "./locales/en/settings.json";
import statisticsEn from "./locales/en/statistics.json";
import ClientListEn from "./locales/en/ClientList.json";
import OfferProfilEn from "./locales/en/OfferProfil.json";
import AgencyHeaderEn from "./locales/en/AgencyHeader.json";
import notificationEn from "./locales/en/notification.json";
import profilEn from "./locales/en/profil.json";


//import profileEn from "./locales/en/profile.json";

const resources = {
  fr: {
    common: commonFr,
    dashboard: dashboardFr,
    offers: offersFr,
    clients: clientsFr,
    settings: settingsFr,
    statistics: statisticsFr,
    ClientList:ClientListFr,
    OfferProfil:OfferProfilFr,
    AgencyHeader:AgencyHeaderFr,
    notification:notificationFr,
    profil:profilFr,


    //profile: profileFr,
  },
  en: {
    common: commonEn,
    dashboard: dashboardEn,
    offers: offersEn,
    clients: clientsEn,
    settings: settingsEn,
    statistics: statisticsEn,
    ClientList:ClientListEn,
    OfferProfil:OfferProfilEn,
    AgencyHeader:AgencyHeaderEn,
    notification:notificationEn,
    profil:profilEn,
    //profile: profileEn,
  },
};

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    resources,
    lng: "fr",                // langue par défaut
    fallbackLng: "fr",        // langue de secours
    ns: ["common","dashboard","offers","clients","settings","statistics","profile"],
    defaultNS: "common",
    interpolation: { escapeValue: false },
  });

export default i18n;
